---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---
#load in of packages
```{r}
pacman::p_load("stringr","gplots","ggplot2","ComplexHeatmap", "limma", "edgeR", "tibble", "Homo.sapiens", "dplyr", "tidyr", "readr", "circlize", "RColorBrewer", "dendextend", "openxlsx","writexl","AnnotationDbi", "clusterProfiler", "org.Hs.eg.db", "tibble","statmod","ggbio", "Mfuzz", "ggupset", "enrichplot", "ComplexUpset", "UpSetR", "gridExtra", "gridtext", "enrichplot", "enrichR", "ggpubr","pracma", "ggh4x","eulerr", "marray", "ggplotify", "corrplot", "cowplot", "forcats", "biomaRt", "DESeq2", "ggpattern", "ssizeRNA", "Polychrome", "devtools", "EnsDb.Hsapiens.v86", "AnnotationHub", "ensembldb","GeneStructureTools","BiocFileCache", "BiocManager", "GO.db", "viridis", "ggridges", "memes", "scales", "ggtext", "htmltools", "drc", "export", "rstatix")
```

#data read in and convertion of ENTREZIDs to ENSEMBL GENEIDS
```{r}
#read in of RNA-seq count matrix

data <- read.table("data/counts_table_20240403.txt", stringsAsFactors = FALSE, check.names = FALSE)

#mapping of ENTREZIDS from count matrix to ENSEMBL GENE IDs

mart <- useMart("ENSEMBL_MART_ENSEMBL")
mart <- useDataset("hsapiens_gene_ensembl", mart)

ens <- rownames(data)
ensLookup <- gsub("\\.[0-9]*$", "", ens)

annot <- getBM(
  mart=mart,
  attributes=c("ensembl_gene_id", "gene_biotype", "external_gene_name"),
  filter="ensembl_gene_id",
  values=ensLookup,
  uniqueRows=TRUE)

annotLookup <- data.frame(
  ens[match(annot$ensembl_gene_id, ensLookup)],
  annot)

colnames(annotLookup) <- c(
  "original_id",
  c("ensembl_gene_id", "gene_biotype", "external_gene_name"))

data$genes <- annotLookup$external_gene_name[match(rownames(data), annotLookup$original_id)]

data$genes <- make.unique(data$genes)

data <- data[!is.na(data$genes) & data$genes != "", ]

rownames(data) <- data$genes

data <- data[,1:353]

#openxlsx::write.xlsx(data,"counts.xlsx", rowNames = TRUE)

counts <- data

#read in of sample information

sampledata <- read.xlsx("data/20230911_sampledata-rnaseq.xlsx")
sampledata <- na.omit(sampledata)
```

#Differential expression analysis using limma
```{r}
group <- t(sampledata[c(3)])
y <- DGEList(counts = counts, group = group)

sex <- sampledata$Sex
sampleid <- sampledata$SAMPLE.ID

y$samples$sex <- as.factor(sex)
y$samples$sampleid <- as.factor(sampleid)

cpm <- cpm(y)
lcpm <- cpm(y, log=TRUE)

L <- mean(y$samples$lib.size) * 1e-6
M <- median(y$samples$lib.size) * 1e-6
c(L, M)
summary(lcpm)

table(rowSums(y$counts==0)==353)
keep.exprs <- filterByExpr(y, group=group)

y <- y[keep.exprs,,keep.lib.sizes=FALSE]
dim(y)

totalcounts <- mean(y$samples$lib.size) *1e-6

y <- calcNormFactors(y, method = "TMM")
y$samples$norm.factors

design <- model.matrix(~0 + group + sex, y$samples)
colnames(design) <- gsub("group", "", colnames(design))

contr.matrix <- makeContrasts(
 "MIDvsPRE" = MID-PRE,
 "POSTvsPRE" = POST-PRE,
 "H3vsPRE" = threeH-PRE,
 "H6vsPRE" = sixH-PRE,
 "H9vsPRE" = nineH-PRE,
 "H12vsPRE" = twelveH-PRE,
 "H24vsPRE" = twentyfourH-PRE,
 "H48vsPRE" = fourtyeightH-PRE,
  levels = colnames(design))
contr.matrix

v <- voom(y, design, plot=TRUE)

#blocking using sample ids to remove batch effects (incl. sex-specific effects) for initial analysis of all participants as a single cohort of 40)

cor <- duplicateCorrelation(counts, design, block=sampleid)

fit <- lmFit(v, design=design, 
  block=sampleid, correlation=cor$consensus.correlation)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
efit <- eBayes(fit)

summary(decideTests(efit, adjust.method = "BH", adj.P.Val=0.05))

rna_comp<- topTable(efit, n=Inf, adjust.method = "BH")
rna_comp <- subset(rna_comp, rna_comp$adj.P.Val<0.05)
openxlsx::write.xlsx(rna_comp,"data/rna_comp.xlsx", rowNames = TRUE)
rna_compALL<- topTable(efit, n=Inf, adjust.method = "BH")
```

#Transcriptomic and protein coding coverage in skeletal muscle (ENCODE & HPA databases) (Fig 3.4A)
```{r}
#GTEx (did not use)

GTExcounts <- read.delim(file="data/gene_reads_muscle_skeletal.gct", skip=2)
GTExtpm <- read.delim(file="data/gene_tpm_muscle_skeletal.gct", skip=2)

rownames(GTExtpm) <- make.names(GTExtpm$Description, unique=TRUE)
GTExtpm <- GTExtpm[,4:806]

rem <- function(x){
  x <- as.matrix(x)
  x <- t(apply(x,1,as.numeric))
  r <- as.numeric(apply(x,1,function(i) sum(i == 0) ))
  remove <- which(r > dim(x)[2]*0.33)
  return(remove)
}
remove <- rem(GTExtpm)

GTExtpmfiltered <- GTExtpm[-remove,]

onlyseqGTE <- as.numeric(length(setdiff(rownames(rna_compALL), rownames(GTExtpmfiltered))))
detectedGTE <- as.numeric(length(intersect(rownames(rna_compALL), rownames(GTExtpmfiltered))))
NotdetectGTE <- as.numeric(length(rownames(GTExtpmfiltered))) - onlyseqGTE - detectedGTE

GTEpiechart <- data.frame(
  category = c("Not in GTEx", "Detected", "Not detected"),
  value = c(onlyseqGTE, detectedGTE , NotdetectGTE)
)

category_colors2 <- c("darkgrey", "white", "grey")

ggplot(GTEpiechart, aes(x = "", y = value, fill = category)) +
  geom_bar(width = 1, stat = "identity") +
  scale_fill_manual(values = category_colors2) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = NULL, fill = "Genotype-Tissue Expression (GTEx) consortium") +
  geom_text(aes(label = paste0(value, "")),
            position = position_stack(vjust = 0.5),
            size = 10) +
    theme(legend.text = element_text(size = 12),
          plot.title = element_text(size = 20))

#ENCODE database for skeletal muscle (date retrieved indicated within the file name)

ENCODE <- read_tsv("data/rna_expression_report_2024_4_26_3h_33m.tsv")

#filtered for only genes with a TPM greater than zero

ENCODE_filtered <- ENCODE[ENCODE$TPM > 0, ]

ENCODE_filtered <- ENCODE_filtered[!duplicated(ENCODE_filtered$`Gene symbol`), ]

DetectonlyinSeq <- as.numeric(length(setdiff(rownames(rna_compALL), ENCODE_filtered$`Gene symbol`)))
detectedENCODE <- as.numeric(length(intersect(rownames(rna_compALL), ENCODE_filtered$`Gene symbol`)))
NotdetectENCODE <- as.numeric(length(ENCODE_filtered$`Gene symbol`)) - DetectonlyinSeq - detectedENCODE

ENCODEpiechart <- data.frame(
  category = c("Detected and not in ENCODE database",
               "Detected and in ENCODE database",
               "Not detected and in ENCODE database"),
  value = c(DetectonlyinSeq, detectedENCODE , NotdetectENCODE))

ENCODEpiechart$percent <- percent(ENCODEpiechart$value / length(as.numeric(rownames(rna_compALL))))

category_colors2 <- c("#cdf7d7", "#e8faed", "lightgrey")

ggplot(ENCODEpiechart, aes(x = "", y = value, fill = category)) +
  geom_bar(width = 1, stat = "identity") +
  scale_fill_manual(values = category_colors2) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = NULL, fill = "Transcriptomic coverage") +
  geom_text(aes(label = paste0(value, "")),
            position = position_stack(vjust = 0.5),
            size = 6) +
  theme(legend.text = element_text(size = 12),
           legend.title = element_text(size = 20, face = "bold"),
            plot.title = element_text(size = 20))

#HPA skeletal muscle database

HPAmuscle <- read_tsv("data/muscleHPA.tsv")
HPAmuscleproteins <- HPAmuscle$Gene

NotdetectHPA <- as.numeric(length(setdiff(HPAmuscleproteins, rownames(rna_compALL))))
detectedHPA <- as.numeric(length(intersect(rownames(rna_compALL), HPAmuscleproteins)))

HPApiechart <- data.frame(
  category = c("Detected and in the Human Protein Atlas",
               "Not detected and in the Human Protein Atlas"),
  value = c(detectedHPA , NotdetectHPA))

category_colors3 <- c("#f29db1", "grey")

ggplot(HPApiechart, aes(x = "", y = value, fill = category)) +
  geom_bar(width = 1, stat = "identity") +
  scale_fill_manual(values = category_colors3) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = NULL, fill = "Protein-encoding gene coverage") +
  geom_text(aes(label = paste0(value, "")),
            position = position_stack(vjust = 0.5),
            size = 6)+
    theme(legend.text = element_text(size = 12),
          legend.title = element_text(size = 20, face = "bold"),
          plot.title = element_text(size = 20),
          plot.margin = ggplot2::margin(0, 1, 0, 0, "cm"))
```

#CV of CPMs at each time point across all participants (3.4B)
```{r}
CVfunction <- function(v) {
  all_genes <- rownames(rna_comp)
  participant_ids <- unique(sampledata$SAMPLE.ID)
  desired_order <- c(5, 2, 4, 7, 6, 3, 8, 9, 1)
      
  features <- data.frame(gene = character(length(all_genes)),
                         timePRECV = numeric(length(all_genes)),
                         timeMIDCV = numeric(length(all_genes)),
                         time0CV = numeric(length(all_genes)),
                         time3CV = numeric(length(all_genes)),
                         time6CV = numeric(length(all_genes)),
                         time9CV = numeric(length(all_genes)),
                         time12CV = numeric(length(all_genes)),
                         time24CV = numeric(length(all_genes)),
                         time48CV = numeric(length(all_genes)),
                         stringsAsFactors = FALSE)
   
  for (i in seq_along(all_genes)) {
    gene <- all_genes[i]
   
    genedata <- as.matrix(tapply(y$counts[gene, ], list(sampleid, group), mean))
    genedata <- genedata[, desired_order]
    genedata <- genedata[, 1:9]
    colnames(genedata) <- c("PRE", "MID", "0", "3", "6", "9", "12", "24", "48")

    timePRE <- genedata[, "PRE"]
    timePRE <- na.omit(timePRE)
    timePRECV <- sd(timePRE)/mean(timePRE)
    timeMID <- genedata[, "MID"]
    timeMID <- na.omit(timeMID)
    timeMIDCV <- sd(timeMID)/mean(timeMID)
    time0 <- genedata[, "0"]
    time0 <- na.omit(time0)
    time0CV <- sd(time0)/mean(time0)
    time3 <- genedata[, "3"]
    time3 <- na.omit(time3)
    time3CV <- sd(time3)/mean(time3)
    time6 <- genedata[, "6"]
    time6 <- na.omit(time6)
    time6CV <- sd(time6)/mean(time6)
    time9 <- genedata[, "9"]
    time9 <- na.omit(time9)
    time9CV <- sd(time9)/mean(time9)
    time12 <- genedata[, "12"]
    time12 <- na.omit(time12)
    time12CV <- sd(time12)/mean(time12)
    time24 <- genedata[, "24"]
    time24 <- na.omit(time24)
    time24CV <- sd(time24)/mean(time24)
    time48 <- genedata[, "48"]
    time48 <- na.omit(time48)
    time48CV <- sd(time48)/mean(time48)

    features[i, ] <- list(gene = gene,
                          timePRECV = timePRECV*100,
                          timeMIDCV = timeMIDCV*100,
                          time0CV = time0CV*100,
                          time3CV = time3CV*100,
                          time6CV = time6CV*100,
                          time9CV = time9CV*100,
                          time12CV = time12CV*100,
                          time24CV= time24CV*100,
                          time48CV = time48CV*100)
    
  }

  return(features)
}

CVcpm <- CVfunction(v)

CVplot <- gather(CVcpm, key = "Time", value = "CV", -gene)
  
label_mapping <- c(
  "timePRECV" = "PRE",
  "timeMIDCV" = "MID",
  "time0CV" = "POST",
  "time3CV" = "3H",
  "time6CV" = "6H",
  "time9CV" = "9H",
  "time12CV" = "12H",
  "time24CV" = "24H",
  "time48CV" = "48H"
)

CVplot$Time <- label_mapping[CVplot$Time]
CVplot$Time <- as.character(CVplot$Time)
CVplot$Time <- factor(CVplot$Time, levels = c("PRE", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H"))

timecolours <-  brewer.pal(9, "Set1")

ggplot(CVplot, aes(x=Time, y = CV, fill=Time)) + 
  scale_fill_manual(values=timecolours, breaks = unique(CVplot$Time)) +
  geom_boxplot(outlier.shape =NA , position = position_dodge(width = 0.8)) + 
  ylim(0,150) +
  labs(x= "Time", y = "Coefficient of variation (%)") +
  theme_bw()

average_CV_by_Time <- CVplot %>%
  group_by(Time) %>%
  summarize(average_CV = median(CV, na.rm = TRUE))

print(average_CV_by_Time)
```

#MDS plot of all samples (BATCHED - block with sample id) (3.4C)
```{r}
mds_data_voom <- plotMDS.DGEList(v)
mds_data_voom <- data.frame(x=mds_data_voom$x,y=mds_data_voom$y,sampledata)

legend_label <- c("PRE" = "PRE", "MID" = "MID", "POST"="POST", "threeH"="3H", "sixH"="6H", "nineH"="9H", "twelveH"="12H", "twentyfourH"="24H", "fourtyeightH"="48H")

ggplot(mds_data_voom,aes(x=x,y=y)) +
  geom_point(aes(color=as.factor(time), shape=as.factor(sex)),size=4) +
  labs(x = "Dimension 1", y = "Dimension 2", colour = "Time point", shape = "Sex") +
  scale_x_continuous(breaks = pretty(mds_data_voom$x, n=14)) +
  scale_y_continuous(breaks = pretty(mds_data_voom$y, n=14)) +
  scale_colour_manual(values = timecolours,
                      labels = legend_label,
                      limits=c("PRE", "MID", "POST", "threeH", "sixH", "nineH", "twelveH", "twentyfourH", "fourtyeightH")) +
  scale_shape_manual(values=c(F=16,M=17),labels=c("F"="Women", "M"="Men")) + 
  theme(axis.title.x = element_text(vjust = 0, size = 20),
        axis.title.y = element_text(vjust = 0, size = 20)) +
  theme(legend.txt = element_text(size =20)) +
  ggtitle("RNA-seq Time Course PCA") +
  geom_rect(xmin = -0.35, xmax = -0.05,
              ymin = 0, ymax = 0.2, size = 1,
              color = "black", alpha = 0) +
  theme(plot.title = element_text(size = 100)) +
  theme_bw()

# zoom-in for the PRE/MID/POST timepoints

zoom <- ggplot(mds_data_voom,aes(x=x,y=y)) +
  geom_point(aes(color=as.factor(time), shape=as.factor(sex)),size=4) +
  labs(x = NULL, y = NULL, colour = NULL) +
  xlim(-0.35, -0.05) + ylim(0, 0.2) +
  scale_colour_manual(values = timecolours, labels = legend_label,
                      limits=c("PRE", "MID", "POST", "threeH", "sixH", "nineH", "twelveH", "twentyfourH", "fourtyeightH")) + 
  scale_shape_manual(values=c(F=16,M=17),labels=c("F"="Women", "M"="Men")) +
  theme_bw()

zoom <- zoom + theme(
  panel.border = element_rect(colour = "black", fill=NA, size=2),
    axis.line = element_blank(),
    axis.text = element_blank(),
    axis.title = element_blank(),
    axis.ticks = element_blank(),
    plot.title = element_blank())

zoom
```

#Overlap plot of previous exercise-induced DEGs (3.5A)
```{r}
#Rubenstein et al. 2022: all DEGs, padj <0.05, RNAseq
rubenstein2022 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/rubenstein2022.xlsx")
rubenstein2022cols <- strsplit(as.character(rubenstein2022$Gene.ID.Gene.Name.baseMean.log2FoldChange.lfcSE.stat.pvalue.padj.Active.Sed.), " ")
rubenstein2022 <- data.frame(do.call(rbind, rubenstein2022cols))
colnames(rubenstein2022) <- c("GeneID", "Symbol", "baseMean", "log2FC", "lfcSE", "stat", "pval", "padj", "ACTIVE", "SED")
rubenstein2022 <- subset(rubenstein2022, rubenstein2022$padj<0.05)
rubenstein2022 <- unique(rubenstein2022$Symbol)

#Popov et al. 2018
#read in of raw data and then DESeq2 as they performed (3 time points, 2 post exercise) (I did not filter genes based on expression by Human Protein Atlas as they did)
popov2018 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/popov2018.xlsx")
popov2018ensembl <- popov2018$Ensembl_gene_id
gene_symbols <- mapIds(org.Hs.eg.db, keys = popov2018ensembl,
                       keytype = "ENSEMBL",
                       column = "SYMBOL") 
gene_symbols <- as.data.frame(gene_symbols)
gene_symbols <- rownames_to_column(gene_symbols, var = "Ensembl_gene_id")
popov2018 <- merge(gene_symbols, popov2018, "Ensembl_gene_id")
popov2018 <- popov2018 %>% dplyr::select(gene_symbols, H0_rep2, H2_rep2, H3_rep2, H0_rep1, H2_rep1, H3_rep1)
popov2018 <- popov2018 %>% dplyr::filter(rowSums(. == 0) == 0)
sum_by_gene <- popov2018 %>%
  group_by(gene_symbols) %>%
  summarize(total_sum = sum(c_across(starts_with("H0_rep2"))))
popov2018 <- popov2018 %>%
  left_join(sum_by_gene, by = "gene_symbols") %>%
  arrange(desc(total_sum)) %>%
  distinct(gene_symbols, .keep_all = TRUE)
rownames(popov2018) <- popov2018$gene_symbols
popov2018 <- popov2018 %>% dplyr::select("H0_rep2", "H2_rep2", "H3_rep2", "H0_rep1", "H2_rep1", "H3_rep1")
popov2018 <- data.frame(apply(popov2018, 2, as.numeric), row.names = rownames(popov2018))
popov2018 <- round(popov2018, digits = 0)

dds <- DESeqDataSetFromMatrix(countData = popov2018,
                              colData = data.frame(time = c("Time1", "Time2", "Time3",
                                                            "Time1", "Time2", "Time3"),
                                                  participant = c(rep("Participant1", 3), 
                                                                  rep("Participant2", 3))),
                                                  design = ~ participant + time)

dds <- DESeq(dds)

res1 <- results(dds, contrast=c("time", "Time2", "Time1"), alpha=0.05)

resOrdered1 <- res1[order(res1$padj),]
write.csv(as.data.frame(resOrdered1), 
          file="condition_treated_results2v1.csv")
popov2v1 <- read_csv("condition_treated_results2v1.csv")
popov2v1 <- subset(popov2v1 , popov2v1$padj<0.05)
popov2v1 <- popov2v1$...1

res2 <- results(dds, contrast=c("time", "Time3", "Time1"), alpha=0.05)

resOrdered2 <- res2[order(res2$padj),]
write.csv(as.data.frame(resOrdered2), 
          file="condition_treated_results3v1.csv")
popov3v1 <- read_csv("condition_treated_results3v1.csv")
popov3v1 <- subset(popov3v1 , popov3v1$padj<0.05)
popov3v1 <- popov3v1$...1

popov2018 <- c(popov2v1, popov3v1)
popov2018 <- unique(popov2018)

#Popov et al. 2019 (same as above, performed DESeq2 on the raw data)
popov2019 <- read.table("data/GSE120862_FPKM.txt", stringsAsFactors = FALSE, check.names = FALSE)
col_names <- as.character(popov2019[1, ])
colnames(popov2019) <- col_names
popov2019 <- popov2019[-1, ]
popov2019 <- popov2019[,2:86]
popov2019 <- popov2019 %>% dplyr::filter(rowSums(. == 0) == 0)
rownames(popov2019) <- make.names(popov2019$gene_name, unique=TRUE)
popov2019 <- popov2019[,-1]
#openxlsx::write.xlsx(popov2019,"popov2019.xlsx", rowNames = TRUE)
cols_to_remove <- grep("NE", names(popov2019))
popov2019 <- popov2019[, -cols_to_remove]
popov2019 <- mutate_all(popov2019, as.numeric)

untrained_cols <- grep("untrained", colnames(popov2019), value = TRUE)
cols_to_remove <- c(untrained_cols)

df_trained <- popov2019[, -which(colnames(popov2019) %in% untrained_cols)]
df_untrained <- popov2019[, untrained_cols]

dds2 <- DESeqDataSetFromMatrix(
  countData = round(df_untrained),
  colData = data.frame(
    time = c("Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3"),
    participant = c(rep("Participant1", 3), rep("Participant2", 3), rep("Participant3", 3), rep("Participant4", 3), rep("Participant5", 3), rep("Participant6", 3), rep("Participant7", 3) )
  ),
  design = ~ participant + time
)

dds2 <- DESeq(dds2)

dds3 <- DESeqDataSetFromMatrix(
  countData = round(df_trained),
  colData = data.frame(
    time = c("Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3", "Time1", "Time2", "Time3",
                    "Time1", "Time2", "Time3"),
    participant = c(rep("Participant1", 3), rep("Participant2", 3), rep("Participant3", 3), rep("Participant4", 3), rep("Participant5", 3), rep("Participant6", 3), rep("Participant7", 3) )
  ),
  design = ~ participant + time
)

dds3 <- DESeq(dds3)

res3 <- results(dds2, contrast=c("time", "Time2", "Time1"), alpha=0.05)
results3 <- as.data.frame(res3@listData)

resOrdered3 <- res3[order(res3$padj),]
write.csv(as.data.frame(resOrdered3), 
          file="popovuntrained2v1.csv")
popov20192v1untrained <- read_csv("popovuntrained2v1.csv")
popov20192v1untrained <- subset(popov20192v1untrained , popov20192v1untrained$padj<0.05)
popov20192v1untrained <- popov20192v1untrained$...1

res4 <- results(dds2, contrast=c("time", "Time3", "Time1"))

resOrdered4 <- res4[order(res4$padj),]
write.csv(as.data.frame(resOrdered4), 
          file="popovuntrained3v1.csv")
popov20193v1untrained <- read_csv("popovuntrained3v1.csv")
popov20193v1untrained <- subset(popov20193v1untrained , popov20193v1untrained$padj<0.05)
popov20193v1untrained <- popov20193v1untrained$...1

res5 <- results(dds3, contrast=c("time", "Time2", "Time1"))

resOrdered5 <- res5[order(res5$padj),]
write.csv(as.data.frame(resOrdered5), 
          file="popovtrained2v1.csv")
popov20192v1trained <- read_csv("popovtrained2v1.csv")
popov20192v1trained <- subset(popov20192v1trained , popov20192v1trained$padj<0.05)
popov20192v1trained <- popov20192v1trained$...1

res6 <- results(dds3, contrast=c("time", "Time3", "Time1"))
results6 <- as.data.frame(res6@listData)

resOrdered6 <- res6[order(res6$padj),]
write.csv(as.data.frame(resOrdered6), 
          file="popovtrained3v1.csv")
popov20193v1trained <- read_csv("popovtrained3v1.csv")
popov20193v1trained <- subset(popov20193v1trained , popov20193v1trained$padj<0.05)
popov20193v1trained <- popov20193v1trained$...1

popov2019 <- c(popov20192v1untrained, popov20193v1untrained, popov20192v1trained, popov20193v1trained)

popov2019 <- unique(popov2019)

#Reitzner et al. 2024
reitzner2024 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/Reitzner2024.xlsx")
reitzner2024 <- reitzner2024 %>% 
  dplyr::filter(factor2 == "EE") %>% dplyr::filter(FDR < 0.05)
reitzner2024 <- unique(reitzner2024$gene)

#This study
taylor2024 <- rownames(rna_comp)

#Pillon et al. 2020: Acute aerobic exercise, all DEG with padj <0.05
pillon2020 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/pillon2020.xlsx")
pillon2020 <- subset(pillon2020, pillon2020$adj.pval_AA<0.05)
pillon2020 <- pillon2020$X1

#Amar et al. 2021: all DEGs impacted by time (FDR of 1% - no total list of genes provided) Meta
amar2021 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/amar2021.xlsx")
amar2021 <- amar2021[grepl('Time-linear|Time-Q', amar2021$Group), ]
amar2021 <- unique(amar2021$Entrez)

#Dickinson et al. 2018: PRE vs 1H and 4H (Aerobic exercise cohort)
dickinson2018 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/dickinson2018.xlsx")
dickinson2018 <- dickinson2018$Gene.Name

#Runqvist et al. 2019, FDR 5%
rundqvist2019 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/rundqvist2019.xlsx")
rundqvist2019 <- rundqvist2019$Symbol

#Moore et al. 2023: 5% FDR using overweight and normal BMI pops. (men) RNAseq
moore2023 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/moore2023.xlsx")
moore2023 <- moore2023$Gene.Symbol

#Norrbom et al. 2022 5% FDR using healthy moderately trained (men) RNAseq
norrbom2022 <- openxlsx::read.xlsx("data/Previousstudies/exerciseresponse/norrbom2022.xlsx")
norrbom2022firstbout <- subset(norrbom2022, norrbom2022$First.bout.of.exercise.adj.P.Val < 0.05)
norrbom2022secondbout <- subset(norrbom2022, norrbom2022$Nineth.bout.of.exercise.adj.P.Val < 0.05)
norrbom2022 <- unique(c(norrbom2022firstbout$hgnc_symbol, norrbom2022secondbout$hgnc_symbol))

previousexercisestudies <- list(
"Rundqvist et al. 2019" = rundqvist2019,
"Pillon et al. 2020" = pillon2020,
"Amar et al. 2021" = amar2021,
"Rubenstein et al. 2022" = rubenstein2022,
"Norrbom et al. 2022" = norrbom2022,
"Moore et al. 2023" = moore2023,
"Reitzner et al. 2024" = reitzner2024,
"Current study" = taylor2024)

UpSetR::upset(fromList(previousexercisestudies), nsets=7, order.by = "freq", nintersects=27,  mb.ratio = c(0.6, 0.4),
      number.angles = 15, 
      text.scale = 1.2, 
      point.size = 2.8, 
      line.size = 1,
      mainbar.y.label = "Overlapping genes",
      sets.x.label = "Number of exercise-induced DEGs")

#creating a list of previously detected genes and newly detected genes
previous_genes <- c(rubenstein2022, rundqvist2019, pillon2020, amar2021, moore2023, popov2018, popov2019, dickinson2018, norrbom2022, reitzner2024)

notdetectedintaylor <- as.data.frame(setdiff(previous_genes, taylor2024))
newgenes <- setdiff(taylor2024, previous_genes)
```

#Overlap plot for exercise-induced DEGs at each time point (Fig 3.5B)
```{r}
#extract individual comparisons
top_treat <- function(coef, efit) {
    result <- topTreat(efit, coef = coef, n = Inf, adjust.method = "BH")
    result <- subset(result, result$adj.P.Val < 0.05)
    return(result)
  }
  
MIDvsPRE <- top_treat(1, efit)
POSTvsPRE <- top_treat(2, efit)
H3vsPRE <- top_treat(3, efit)
H6vsPRE <- top_treat(4, efit)
H9vsPRE <- top_treat(5, efit)
H12vsPRE <- top_treat(6, efit)
H24vsPRE <- top_treat(7, efit)
H48vsPRE <- top_treat(8, efit)
  
  
overlap_genes <- list(
  MID = rownames(MIDvsPRE),
  POST = rownames(POSTvsPRE),
  H3 = rownames(H3vsPRE),
  H6 = rownames(H6vsPRE),
  H9 = rownames(H9vsPRE),
  H12 = rownames(H12vsPRE),
  H24 = rownames(H24vsPRE),
  H48 = rownames(H48vsPRE))

all_genes <- unlist(overlap_genes)
unique_genes <- unique(all_genes)

count_lists_appeared <- function(gene, gene_lists) {
  sum(sapply(gene_lists, function(lst) gene %in% lst))}

gene_counts <- sapply(unique_genes, function(gene) count_lists_appeared(gene, overlap_genes))

gene_matrix <- sapply(overlap_genes, function(genes) {
  sapply(unique_genes, function(gene) as.numeric(gene %in% genes))})

gene_counts_summary <- table(rowSums(gene_matrix))

timecolours2 <-  brewer.pal(8, "Set2")

pie(gene_counts_summary, 
    labels = NA, 
    main = "Overlap of DEGs relative to PRE",
    col = timecolours,
    cex = 0.8)

legend("topright", legend = paste(names(gene_counts_summary),
                                  "time point(s):", as.vector(gene_counts_summary), "genes"), 
       cex = 0.8, 
       fill = timecolours2,
       inset = c(0.05, 0.05),
       bg = "transparent")

names(overlap_genes) = c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")

time_palettecustom2 <- timecolours[-c(1)]

#Overlap plot for DE expression across time points
UpSetR::upset(fromList(overlap_genes), sets = c("48H", "24H", "12H", "9H", "6H", "3H", "POST", "MID"),
      nsets=8, order.by = "freq",
      nintersects=25,
      mb.ratio = c(0.6, 0.4),
      keep.order=T,
      number.angles = 15,
      text.scale = 1.2, 
      point.size = 2.8, 
      line.size = 1,
      sets.bar.color = rev(time_palettecustom2),
      mainbar.y.label = "Overlapping genes",
      sets.x.label = "Number of DEGs relative to PRE")

  MIDvsPREdown <- MIDvsPRE %>% dplyr::filter(logFC < 0)
  MIDvsPREup <- MIDvsPRE %>% dplyr::filter(logFC > 0)
  POSTvsPREdown <- POSTvsPRE %>% dplyr::filter(logFC < 0)
  POSTvsPREup <- POSTvsPRE %>% dplyr::filter(logFC > 0)
  H3vsPREdown <- H3vsPRE %>% dplyr::filter(logFC < 0)
  H3vsPREup <- H3vsPRE %>% dplyr::filter(logFC > 0)
  H6vsPREdown <- H6vsPRE %>% dplyr::filter(logFC < 0)
  H6vsPREup <- H6vsPRE %>% dplyr::filter(logFC > 0)
  H9vsPREdown <- H9vsPRE %>% dplyr::filter(logFC < 0)
  H9vsPREup <- H9vsPRE %>% dplyr::filter(logFC > 0)
  H12vsPREdown <- H12vsPRE %>% dplyr::filter(logFC < 0)
  H12vsPREup <- H12vsPRE %>% dplyr::filter(logFC > 0)
  H24vsPREdown <- H24vsPRE %>% dplyr::filter(logFC < 0)
  H24vsPREup <- H24vsPRE %>% dplyr::filter(logFC > 0)
  H48vsPREdown <- H48vsPRE %>% dplyr::filter(logFC < 0)
  H48vsPREup <- H48vsPRE %>% dplyr::filter(logFC > 0)
  
  
overlapdirection_genes <- list(
  MIDup = rownames(MIDvsPREup),
  POSTup = rownames(POSTvsPREup),
  H3up = rownames(H3vsPREup),
  H6up = rownames(H6vsPREup),
  H9up = rownames(H9vsPREup),
  H12up = rownames(H12vsPREup),
  H24up = rownames(H24vsPREup),
  H48up = rownames(H48vsPREup),
  MIDdown = rownames(MIDvsPREdown),
  POSTdown = rownames(POSTvsPREdown),
  H3down = rownames(H3vsPREdown),
  H6down = rownames(H6vsPREdown),
  H9down = rownames(H9vsPREdown),
  H12down = rownames(H12vsPREdown),
  H24down = rownames(H24vsPREdown),
  H48down = rownames(H48vsPREdown))

names(overlapdirection_genes) = c("MID UP", "POST UP", "3H UP", "6H UP", "9H UP", "12H UP", "24H UP", "48H UP", "MID DOWN", "POST DOWN", "3H DOWN", "6H DOWN", "9H DOWN", "12H DOWN", "24H DOWN", "48H DOWN")

UpSetR::upset(fromList(overlapdirection_genes), sets = c("MID UP", "POST UP", "3H UP", "6H UP", "9H UP", "12H UP", "24H UP", "48H UP", "MID DOWN", "POST DOWN", "3H DOWN", "6H DOWN", "9H DOWN", "12H DOWN", "24H DOWN", "48H DOWN"),
      nsets=16, order.by = "freq",
      nintersects=50,
      mb.ratio = c(0.6, 0.4),
      keep.order=T,
      number.angles = 15, 
      text.scale = 1.4, 
      point.size = 2.8, 
      line.size = 1)
```

#Plot of no. DEGs and average expression at each time point, with previous exercise-induced DEGS indicated through col pattern (3.5C)
```{r}
process_DE_genes <- function(data, time_point, previous_genes) {
  
  DE_down <- data %>% dplyr::filter(logFC < 0)
  meanlogFCdown <- as.numeric(mean(DE_down$logFC))
  sdlogFCdown <-  as.numeric(sd(DE_down$logFC))
  DE_down <- rownames(DE_down)
  
  DE_down_previous <- intersect(DE_down, previous_genes)
  DE_down_previous <- length(DE_down_previous)
  DE_down_previous <- DE_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Down", Time = time_point, Previous = "Previously identified") %>%
    cbind(meanlogFCdown) %>% cbind(sdlogFCdown)
  colnames(DE_down_previous) <- c("Genes", "Direction", "Time", "Previous", "meanlogFC", "sdlogFC")
  DE_down_previous$Genes <- -DE_down_previous$Genes
  
  DE_down_new <- setdiff(DE_down, previous_genes)
  DE_down_new <- length(DE_down_new)
  DE_down_new <- DE_down_new %>% as.data.frame() %>%
    mutate(Direction = "Down", Time = time_point, Previous = "Never") %>%
    cbind(meanlogFCdown) %>% cbind(sdlogFCdown)
  colnames(DE_down_new) <- c("Genes", "Direction", "Time", "Previous", "meanlogFC", "sdlogFC")
  DE_down_new$Genes <- -DE_down_new$Genes
  
  DE_up <- data %>% dplyr::filter(logFC > 0)
  meanlogFCup <- as.numeric(mean(DE_up$logFC))
  sdlogFCup <-  as.numeric(sd(DE_up$logFC))
  DE_up <- rownames(DE_up)
  
  DE_up_previous <- intersect(DE_up, previous_genes)
  DE_up_previous <- length(DE_up_previous)
  DE_up_previous <- DE_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Up", Time = time_point, Previous = "Previously identified") %>%
    cbind(meanlogFCup) %>% cbind(sdlogFCup)
  colnames(DE_up_previous) <- c("Genes", "Direction", "Time", "Previous", "meanlogFC", "sdlogFC")
  
  DE_up_new <- setdiff(DE_up, previous_genes)
  DE_up_new <- length(DE_up_new)
  DE_up_new <- DE_up_new %>% as.data.frame() %>%
    mutate(Direction = "Up", Time = time_point, Previous = "Never") %>%
    cbind(meanlogFCup) %>% cbind(sdlogFCup)
  colnames(DE_up_new) <- c("Genes", "Direction", "Time", "Previous", "meanlogFC", "sdlogFC")
  
  DE_data <- rbind(DE_down_previous, DE_up_previous, DE_down_new, DE_up_new)
  return(DE_data)
}

TotalDE1 <- process_DE_genes(MIDvsPRE, "MID", previous_genes)
TotalDE2 <- process_DE_genes(POSTvsPRE, "POST", previous_genes)
TotalDE3 <- process_DE_genes(H3vsPRE, "3H", previous_genes)
TotalDE4 <- process_DE_genes(H6vsPRE, "6H", previous_genes)
TotalDE5 <- process_DE_genes(H9vsPRE, "9H", previous_genes)
TotalDE6 <- process_DE_genes(H12vsPRE, "12H", previous_genes)
TotalDE7 <- process_DE_genes(H24vsPRE, "24H", previous_genes)
TotalDE8 <- process_DE_genes(H48vsPRE, "48H", previous_genes)

TotalDE <- rbind (TotalDE1, TotalDE2, TotalDE3, TotalDE4, TotalDE5, TotalDE6, TotalDE7, TotalDE8)

TotalDE$Time <- as.character(TotalDE$Time)
TotalDE$Time <- factor(TotalDE$Time, levels = unique(TotalDE$Time))

time_palettecustom2 <- timecolours[-c(1)]
time_palettecustom2dark <- c("#991010", "#1E4E73", "#397833",
                             "#66486A", "#CC6600", "#CCCC26",
                             "#80421D", "#C25E9F", "#666666")
time_palettecustom2dark <- time_palettecustom2dark[-c(1)]

TotalDE <- TotalDE %>%
  mutate(Time_Direction = paste(Time, Direction, sep = "_"))

ggplot(data = TotalDE, aes(x = Time, y = Genes, fill = Time_Direction, pattern = Previous)) + 
  geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
  labs(x = "Time", y = "No. DEGs relative to PRE", fill = "Direction",
       pattern = str_wrap("Previously identified as exercise-induced DEG", width = 25)) +
  scale_fill_manual(values = c("MID_Up" = time_palettecustom2[1],
                               "MID_Down" = time_palettecustom2dark[1],
                               "POST_Up" = time_palettecustom2[2],
                               "POST_Down" = time_palettecustom2dark[2],
                               "3H_Up" = time_palettecustom2[3],
                               "3H_Down" = time_palettecustom2dark[3],
                               "6H_Up" = time_palettecustom2[4],
                               "6H_Down" = time_palettecustom2dark[4],
                               "9H_Up" = time_palettecustom2[5],
                               "9H_Down" = time_palettecustom2dark[5],
                               "12H_Up" = time_palettecustom2[6],
                               "12H_Down" = time_palettecustom2dark[6],
                               "24H_Up" = time_palettecustom2[7],
                               "24H_Down" = time_palettecustom2dark[7],
                               "48H_Up" = time_palettecustom2[8],
                               "48H_Down" = time_palettecustom2dark[8])) + 
  scale_pattern_manual(values = c("Never" = "none", "Previously identified" = "stripe"),
                       labels = c("Previously identified" = "Yes", "Never" = "No")) +
  geom_point(data = TotalDE, aes(x = Time, y = meanlogFC * 2000, shape = Direction), size = 4) +
  geom_errorbar(data = TotalDE, 
                aes(ymin = (meanlogFC - sdlogFC) * 2000,
                    ymax = (meanlogFC + sdlogFC) * 2000),
                width = 0.5, size = 0.75) +
  scale_y_continuous(name = expression("        No. DEGs relative to PRE\n \n Downregulated                Upregulated"), 
        sec.axis = sec_axis(~ . * 0.0005,
                        name =  bquote("Average" ~ log[2] ~ "FC of DEGs relative to PRE"),
                        breaks = c(-3, -2.5, -2, -1.5, -1, -0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3)), 
                        labels = function(x) abs(x),
                        limits = c(-6000, 6000),
                        breaks = seq(-6000, 6000, by = 1000)) +
  scale_shape_manual(name = "Average DEG fold change",
                   values = c(Up = 15, Down = 18),
                   labels = c("Up" = "Upregulated genes", "Down" = "Downregulated genes")) +
  theme_bw() +
  theme(axis.title.y = element_text(margin = ggplot2::margin(r = 5, l = 20)),
        axis.title.y.right = element_text(margin = ggplot2::margin(l = 5, r =5))) +
    ggtitle("Exercise-induced DEGs") +
    guides( 
    fill = "none",
    pattern = guide_legend(reverse = TRUE, override.aes = list(fill = "white")),
    shape = guide_legend(reverse = TRUE))
```

#Unsupervised hierarchical clustering to create the heatmap of all DEGs (3.5D)
```{r}
dendrowgraph <- rna_comp[c(1:8)] %>% dist(method = "euclidean") %>%
  hclust(method = "average") %>% as.dendrogram

col <- colorRamp2(c(-2, 0, 2), c("#0A3C72", "white", "#7B0722"))

labelsHM <- expression(Log[2]*FC ~ "relative to PRE")

heatmap <- Heatmap(
  as.matrix(rna_comp[c(1, 2, 3, 4, 5, 6, 7, 8)]), 
  heatmap_legend_param = list(
    title = labelsHM, 
    legend_width = unit(3.6, "cm"),
    legend_direction = "horizontal"),
  col = col, 
  cluster_rows = dendrowgraph, 
  show_column_dend = TRUE,
  show_row_names = FALSE, 
  show_row_dend = FALSE, 
  cluster_columns = FALSE, 
  column_names_gp = gpar(fontsize = 10), 
  column_labels = c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H"), 
  column_names_rot = 45)

draw(heatmap, heatmap_legend_side = "bottom")
```

#generating dataframe of correlations of differential gene expression to baseline VO2
```{r}
vo2 <- sampledata %>%
  distinct(SAMPLE.ID, .keep_all = TRUE)
vo2 <- vo2[-c(12,26), ]
vo2list <- vo2$vo2

VO2correlations <- function(v) {
  all_genes <- rownames(rna_comp)
  participant_ids <- unique(vo2$SAMPLE.ID)
  desired_order <- c(5, 2, 4, 7, 6, 3, 8, 9, 1)
  
  MIDvsPREUp <- MIDvsPRE[MIDvsPRE$logFC > 0,]
  POSTvsPREUp <- POSTvsPRE[POSTvsPRE$logFC > 0,]
  H3vsPREUp <- H3vsPRE[H3vsPRE$logFC > 0,]
  H6vsPREUp <- H6vsPRE[H6vsPRE$logFC > 0,]
  H9vsPREUp <- H9vsPRE[H9vsPRE$logFC > 0,]
  H12vsPREUp <- H12vsPRE[H12vsPRE$logFC > 0,]
  H24vsPREUp <- H24vsPRE[H24vsPRE$logFC > 0,]
  H48vsPREUp <- H48vsPRE[H48vsPRE$logFC > 0,]
  
  MIDvsPREDown <- MIDvsPRE[MIDvsPRE$logFC < 0,]
  POSTvsPREDown <- POSTvsPRE[POSTvsPRE$logFC < 0,]
  H3vsPREDown <- H3vsPRE[H3vsPRE$logFC < 0,]
  H6vsPREDown <- H6vsPRE[H6vsPRE$logFC < 0,]
  H9vsPREDown <- H9vsPRE[H9vsPRE$logFC < 0,]
  H12vsPREDown <- H12vsPRE[H12vsPRE$logFC < 0,]
  H24vsPREDown <- H24vsPRE[H24vsPRE$logFC < 0,]
  H48vsPREDown <- H48vsPRE[H48vsPRE$logFC < 0,]
  
# Initialize a data frame to store features for each gene
features <- data.frame(gene = character(length(all_genes)),
                          feature_correlation_peakfcup_feature = numeric(length(all_genes)),
                          feature_pvalue_peakfcup_feature = numeric(length(all_genes)),
                          feature_correlation_peakfcdown_feature = numeric(length(all_genes)),
                          feature_pvalue_peakfcdown_feature = numeric(length(all_genes)),
                          feature_correlation_avgfc_feature = numeric(length(all_genes)),
                          feature_pvalue_avgfc_feature = numeric(length(all_genes)),
                          feature_correlation_genestdev_feature = numeric(length(all_genes)),
                          feature_pvalue_genestdev_feature = numeric(length(all_genes)),
                          feature_correlation_auc_feature = numeric(length(all_genes)),
                          feature_pvalue_auc_feature = numeric(length(all_genes)),
                          Up_correlation_tMID_feature = numeric(length(all_genes)),
                          Up_pvalue_tMID_feature = numeric(length(all_genes)),
                          Up_correlation_t0_feature = numeric(length(all_genes)),
                          Up_pvalue_t0_feature = numeric(length(all_genes)),
                          Up_correlation_t3_feature = numeric(length(all_genes)),
                          Up_pvalue_t3_feature = numeric(length(all_genes)),
                          Up_correlation_t6_feature = numeric(length(all_genes)),
                          Up_pvalue_t6_feature = numeric(length(all_genes)),
                          Up_correlation_t9_feature = numeric(length(all_genes)),
                          Up_pvalue_t9_feature = numeric(length(all_genes)),
                          Up_correlation_t12_feature = numeric(length(all_genes)),
                          Up_pvalue_t12_feature = numeric(length(all_genes)),
                          Up_correlation_t24_feature = numeric(length(all_genes)),
                          Up_pvalue_t24_feature = numeric(length(all_genes)),
                          Up_correlation_t48_feature = numeric(length(all_genes)),
                          Up_pvalue_t48_feature = numeric(length(all_genes)),
                          Down_correlation_tMID_feature = numeric(length(all_genes)),
                          Down_pvalue_tMID_feature = numeric(length(all_genes)),
                          Down_correlation_t0_feature = numeric(length(all_genes)),
                          Down_pvalue_t0_feature = numeric(length(all_genes)),
                          Down_correlation_t3_feature = numeric(length(all_genes)),
                          Down_pvalue_t3_feature = numeric(length(all_genes)),
                          Down_correlation_t6_feature = numeric(length(all_genes)),
                          Down_pvalue_t6_feature = numeric(length(all_genes)),
                          Down_correlation_t9_feature = numeric(length(all_genes)),
                          Down_pvalue_t9_feature = numeric(length(all_genes)),
                          Down_correlation_t12_feature = numeric(length(all_genes)),
                          Down_pvalue_t12_feature = numeric(length(all_genes)),
                          Down_correlation_t24_feature = numeric(length(all_genes)),
                          Down_pvalue_t24_feature = numeric(length(all_genes)),
                          Down_correlation_t48_feature = numeric(length(all_genes)),
                          Down_pvalue_t48_feature = numeric(length(all_genes)),
                         stringsAsFactors = FALSE)
   
  # Loop through each gene
  for (i in seq_along(all_genes)) {
    gene <- all_genes[i]
  
    # Get gene expression data for the current gene
    genedata <- as.matrix(tapply(v$E[gene, ], list(sampleid, group), mean))
    genedata <- genedata[, desired_order]
    genedata[, 1:9] <- genedata[, 1:9] - genedata[, 1]
    genedata <- genedata[, 2:9]
    colnames(genedata) <- c("MID", "0", "3", "6", "9", "12", "24", "48")
    rows_to_remove <- c(12,26)
    genedata <- genedata[-rows_to_remove, ]

    # Calculate features
    peakfcup <- apply(genedata, 1, max)
    peakfcdown <- apply(genedata, 1, min)
    avgfc <- rowMeans(genedata)
    genestdev <- apply(genedata, 1, sd)
    timeMID <- genedata[, "MID"]
    time0 <- genedata[, "0"]
    time3 <- genedata[, "3"]
    time6 <- genedata[, "6"]
    time9 <- genedata[, "9"]
    time12 <- genedata[, "12"]
    time24 <- genedata[, "24"]
    time48 <- genedata[, "48"]
    
    # Calculate AUC using the trapezoidal rule
    calculate_auc_trapz <- function(row) {
      x <- as.numeric(colnames(genedata))[-1] 
      y <- as.numeric(row[-1])                
      auc <- trapz(x, y)
      return(auc)
    }
    auc <- apply(genedata, 1, calculate_auc_trapz)
    auc <- abs(auc)
  
    # Calculate correlations with feature and their p-values
    cor_test <- function(x, y) {
      cor_val <- cor.test(x, y)$estimate
      p_val <- cor.test(x, y)$p.value
      return(list(cor_val, p_val))
    }
    
    c_peakfcup <- cor_test(peakfcup, vo2list)
    c_peakfcdown <- cor_test(peakfcdown, vo2list)
    c_avgfc <- cor_test(avgfc, vo2list)
    c_genestdev <- cor_test(genestdev, vo2list)
    c_auc <- cor_test(auc, vo2list)
    
#only keep genes that are differentially expressed at time point a correlation is detected
c_MIDUp <- if (gene %in% rownames(MIDvsPREUp)) {
  cor_test(timeMID, vo2list)
} else {
  list(NA, NA)
}

c_t0Up <- if (gene %in% rownames(POSTvsPREUp)) {
  cor_test(time0, vo2list)
} else {
  list(NA, NA)
}
    
    
c_t3Up <- if (gene %in% rownames(H3vsPREUp)) {
  cor_test(time3, vo2list)
} else {
  list(NA, NA)
}

c_t6Up <- if (gene %in% rownames(H6vsPREUp)) {
  cor_test(time6, vo2list)
} else {
  list(NA, NA)
}

c_t9Up <- if (gene %in% rownames(H9vsPREUp)) {
  cor_test(time9, vo2list)
} else {
  list(NA, NA)
}

c_t12Up <- if (gene %in% rownames(H12vsPREUp)) {
  cor_test(time12, vo2list)
} else {
  list(NA, NA)
}

c_t24Up <- if (gene %in% rownames(H24vsPREUp)) {
  cor_test(time24, vo2list)
} else {
  list(NA, NA)
}

c_t48Up <- if (gene %in% rownames(H48vsPREUp)) {
  cor_test(time48, vo2list)
} else {
  list(NA, NA)
}

c_MIDDown <- if (gene %in% rownames(MIDvsPREDown)) {
  cor_test(timeMID, vo2list)
} else {
  list(NA, NA)
}

c_t0Down <- if (gene %in% rownames(POSTvsPREDown)) {
  cor_test(time0, vo2list)
} else {
  list(NA, NA)
}
    
    
c_t3Down <- if (gene %in% rownames(H3vsPREDown)) {
  cor_test(time3, vo2list)
} else {
  list(NA, NA)
}

c_t6Down <- if (gene %in% rownames(H6vsPREDown)) {
  cor_test(time6, vo2list)
} else {
  list(NA, NA)
}

c_t9Down <- if (gene %in% rownames(H9vsPREDown)) {
  cor_test(time9, vo2list)
} else {
  list(NA, NA)
}

c_t12Down <- if (gene %in% rownames(H12vsPREDown)) {
  cor_test(time12, vo2list)
} else {
  list(NA, NA)
}

c_t24Down <- if (gene %in% rownames(H24vsPREDown)) {
  cor_test(time24, vo2list)
} else {
  list(NA, NA)
}

c_t48Down <- if (gene %in% rownames(H48vsPREDown)) {
  cor_test(time48, vo2list)
} else {
  list(NA, NA)
}
    # Store results for the current gene in the data frame
    features[i, ] <- list(gene = gene,
                          feature_correlation_peakfcup_feature = c_peakfcup[[1]],
                          feature_pvalue_peakfcup_feature = c_peakfcup[[2]],
                          feature_correlation_peakfcdown_feature = c_peakfcdown[[1]],
                          feature_pvalue_peakfcdown_feature = c_peakfcdown[[2]],
                          feature_correlation_avgfc_feature = c_avgfc[[1]],
                          feature_pvalue_avgfc_feature = c_avgfc[[2]],
                          feature_correlation_genestdev_feature = c_genestdev[[1]],
                          feature_pvalue_genestdev_feature = c_genestdev[[2]],
                          feature_correlation_auc_feature = c_auc[[1]],
                          feature_pvalue_auc_feature = c_auc[[2]],
                          Up_correlation_tMID_feature = c_MIDUp[[1]],
                          Up_pvalue_tMID_feature = c_MIDUp[[2]],
                          Up_correlation_t0_feature = c_t0Up[[1]],
                          Up_pvalue_t0_feature = c_t0Up[[2]],
                          Up_correlation_t3_feature = c_t3Up[[1]],
                          Up_pvalue_t3_feature = c_t3Up[[2]],
                          Up_correlation_t6_feature = c_t6Up[[1]],
                          Up_pvalue_t6_feature = c_t6Up[[2]],
                          Up_correlation_t9_feature = c_t9Up[[1]],
                          Up_pvalue_t9_feature = c_t9Up[[2]],
                          Up_correlation_t12_feature = c_t12Up[[1]],
                          Up_pvalue_t12_feature = c_t12Up[[2]],
                          Up_correlation_t24_feature = c_t24Up[[1]],
                          Up_pvalue_t24_feature = c_t24Up[[2]],
                          Up_correlation_t48_feature = c_t48Up[[1]],
                          Up_pvalue_t48_feature = c_t48Up[[2]],
                          Down_correlation_tMID_feature = c_MIDDown[[1]],
                          Down_pvalue_tMID_feature = c_MIDDown[[2]],
                          Down_correlation_t0_feature = c_t0Down[[1]],
                          Down_pvalue_t0_feature = c_t0Down[[2]],
                          Down_correlation_t3_feature = c_t3Down[[1]],
                          Down_pvalue_t3_feature = c_t3Down[[2]],
                          Down_correlation_t6_feature = c_t6Down[[1]],
                          Down_pvalue_t6_feature = c_t6Down[[2]],
                          Down_correlation_t9_feature = c_t9Down[[1]],
                          Down_pvalue_t9_feature = c_t9Down[[2]],
                          Down_correlation_t12_feature = c_t12Down[[1]],
                          Down_pvalue_t12_feature = c_t12Down[[2]],
                          Down_correlation_t24_feature = c_t24Down[[1]],
                          Down_pvalue_t24_feature = c_t24Down[[2]],
                          Down_correlation_t48_feature = c_t48Down[[1]],
                          Down_pvalue_t48_feature = c_t48Down[[2]])
  }
  
  return(features)
}

VO2correlated <- VO2correlations(v)

VO2correlatedfilter <- VO2correlated %>%
  pivot_longer(
    cols = -gene,
    names_to = "variable",
    values_to = "value"
  ) %>%
  separate(variable, into = c("regulation", "type", "index"), sep = "_")

VO2correlatedfilter <- VO2correlatedfilter %>%
  pivot_wider(names_from = type, values_from = value)

VO2correlatedfilter <- VO2correlatedfilter %>% dplyr::filter(pvalue < 0.05)

VO2correlatedfilter <- VO2correlatedfilter %>%
  mutate(direction = ifelse(correlation < 0, "negative", "positive"))

ggplot(VO2correlatedfilter, aes(x=index, y=correlation)) +
  geom_violin(stat = "ydensity", scale = "count") +
  scale_colour_manual() +
  theme_set(theme_bw())

ggplot(VO2correlatedfilter, aes(x = correlation, y = as.factor(index))) +
  geom_density_ridges() +
  labs(x = "Correlation", y = "Index") +
  theme_set(theme_bw())
```

#no. DEGs correlated to vo2 for timepoints (3.6A)
```{r}
correlationcount <- VO2correlatedfilter %>%
  group_by(index,direction,regulation) %>%
  summarize(gene_count = n_distinct(gene))

totalgenes <- data.frame(totalgene = c(as.numeric(nrow(MIDvsPRE)), as.numeric(nrow(POSTvsPRE)), as.numeric(nrow(H3vsPRE)), as.numeric(nrow(H6vsPRE)), as.numeric(nrow(H9vsPRE)), as.numeric(nrow(H12vsPRE)), as.numeric(nrow(H24vsPRE)), as.numeric(nrow(H48vsPRE)), 15024, 15024, 15024, 15024, 15024),
index = c("tMID", "t0", "t3", "t6", "t9", "t12", "t24", "t48", "peakfcup", "peakfcdown", "avgfc", "genestdev", "auc"))

merged_df <- merge(correlationcount, totalgenes, by = "index")

merged_df$gene_count <- ifelse(merged_df$regulation == "Down",
                               merged_df$gene_count * -1,
                               merged_df$gene_count)

merged_df$percentage <- (merged_df$gene_count / merged_df$totalgene) *100 

timepointcorrelation <- merged_df %>%
  dplyr::filter(index %in% c("tMID", "t0", "t3", "t6", "t9", "t12", "t24", "t48"))

timepointcorrelation$index <- factor(timepointcorrelation$index, levels = c("tMID", "t0", "t3", "t6", "t9", "t12", "t24", "t48"))

levels(timepointcorrelation$index) <- c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")

timepointcorrelation <- timepointcorrelation %>%
  mutate(Time_Regulation = paste(index, regulation, sep = "_"))

time_palettecustom2 <- timecolours[-c(1)]
time_palettecustom2dark <- c("#991010", "#1E4E73", "#397833", "#66486A", "#CC6600", "#CCCC26", "#80421D", "#C25E9F", "#666666")
time_palettecustom2dark <- time_palettecustom2dark[-c(1)]

timepointcorrelation <- timepointcorrelation %>%
 mutate(
    effect = case_when(
      direction == "positive" & regulation == "Down" ~ "Greater in lower fitness",
      direction == "positive" & regulation == "Up" ~ "Greater in higher fitness",
      direction == "negative" & regulation == "Down" ~ "Greater in higher fitness",
      direction == "negative" & regulation == "Up" ~ "Greater in lower fitness",
      TRUE ~ NA_character_
    )
  )

ggplot(timepointcorrelation, aes(x = index, y = percentage, fill = Time_Regulation, pattern = effect)) +
  geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
  scale_pattern_manual(values = c("Greater in lower fitness" = "none", "Greater in higher fitness" = "circle"),
        labels = c("Greater in lower fitness" = "Greater in lower fitness", "Greater in higher fitness" = "Greater in higher fitness")) +
  scale_y_continuous(name = expression(atop("   Proportion of DEGs correlated to VO"[2*peak]~" (%)", " \n \n Downregulated                        Upregulated")), labels = function(x) abs(x), 
                     limits= c(-25, 25),
                     breaks=c(-25, 25, by = 10, -10,15,-15,20,-20,5,-5, 0)) +
  scale_fill_manual(values = c(
    "MID_Up" = time_palettecustom2[1], "MID_Down" = time_palettecustom2dark[1],
    "POST_Up" = time_palettecustom2[2], "POST_Down" = time_palettecustom2dark[2],
    "3H_Up" = time_palettecustom2[3], "3H_Down" = time_palettecustom2dark[3],
    "6H_Up" = time_palettecustom2[4], "6H_Down" = time_palettecustom2dark[4],
    "9H_Up" = time_palettecustom2[5], "9H_Down" = time_palettecustom2dark[5],
    "12H_Up" = time_palettecustom2[6], "12H_Down" = time_palettecustom2dark[6],
    "24H_Up" = time_palettecustom2[7], "24H_Down" = time_palettecustom2dark[7],
    "48H_Up" = time_palettecustom2[8], "48H_Down" = time_palettecustom2dark[8])) + 
  labs(x = NULL, fill = "Time point", pattern = "Magnitude of differential expression") +
  ggtitle(bquote("Correlation between exercise-induced DEG expression and baseline" ~ VO[2*"peak"])) +
  theme(axis.title.y = element_text(margin = ggplot2::margin(r = -5))) +
  guides(fill = FALSE,
         pattern = guide_legend(reverse = TRUE,
                                override.aes = list(fill = "white"))) +
  theme_bw()
```

#no. DEGs correlated to VO2 for gene features (3.6B)
```{r}
featurecorrelation <- merged_df %>%
  dplyr::filter(index %in% c("peakfcup", "genestdev", "peakfcdown", "auc", "avgfc"))

featurecorrelation$index <- factor(featurecorrelation$index, levels = c("peakfcup", "peakfcdown", "avgfc", "genestdev", "auc"))

levels(featurecorrelation$index) <- c("Peak FC up", "Peak FC down", "Average FC", "Standard deviation", "AUC")

featurecolours <-  brewer.pal(5, "Set3")

ggplot(featurecorrelation, aes(x = index, y = percentage, fill = index, pattern = direction)) + 
  geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
  scale_pattern_manual(values = c("positive" = "circle","negative" = "none"),
                       labels = c( "negative" = "Greater in lower fitness",
                                   "positive" = "Greater in higher fitness")) +
  scale_fill_manual(values = featurecolours) +
  scale_y_continuous(name = expression("Proportion of DEGs correlated to VO"[2*peak]~"(%)"),
                     expand = expansion(mult = c(0, 0.05)), limits= c(0, 50)) +
  labs(x = NULL, fill = "Expression feature", pattern = "Magnitude of differential expression") + 
  ggtitle("Correlation between exercise-induced DEG expression and baseline" ~ VO[2*"peak"]) +
  theme(axis.title.y = element_text(margin = ggplot2::margin(r = -5))) +
  guides(fill = FALSE,
         pattern = guide_legend(reverse = TRUE,
                  override.aes = list(fill = "white"))) +
  theme_bw()
```

#enrichment of correlation between vo2 and genes at timepoints (3.6C)
```{r}
timeVO2_enrichGO <- function(time_point, directionVO2, regulationVO2, comparison) {
  
  correlatedVO2 <- VO2correlatedfilter %>%
  dplyr::filter(index == time_point & direction == directionVO2 & regulation == regulationVO2) %>%
  dplyr::pull(gene)

  correlatedVO2 <- enrichGO(correlatedVO2, OrgDb = "org.Hs.eg.db",
                                 universe=rownames(comparison), keyType = "SYMBOL",
                                 ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.2,
                                 pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

  correlatedVO2 <- as.data.frame(correlatedVO2@result) %>% arrange(p.adjust) %>%
  dplyr::filter(p.adjust < 0.05) %>% head(5) %>%
  mutate(Index = time_point, Direction = directionVO2, Regulation = regulationVO2)
  
  return(correlatedVO2)
}

H3correlationposUp <- timeVO2_enrichGO("t3", "positive", "Up", H3vsPRE)
H6correlationposUp <- timeVO2_enrichGO("t6", "positive", "Up", H6vsPRE)
H9correlationposUp <- timeVO2_enrichGO("t9", "positive", "Up", H9vsPRE)
H12correlationposUp <- timeVO2_enrichGO("t12", "positive", "Up", H12vsPRE)
H24correlationposUp <- timeVO2_enrichGO("t24", "positive", "Up", H24vsPRE)
H48correlationposUp <- timeVO2_enrichGO("t48", "positive", "Up", H48vsPRE)
H3correlationnegUp <- timeVO2_enrichGO("t3", "negative", "Up", H3vsPRE)
H6correlationnegUp <- timeVO2_enrichGO("t6", "negative", "Up", H6vsPRE)
H9correlationnegUp <- timeVO2_enrichGO("t9", "negative", "Up", H9vsPRE)
H12correlationnegUp <- timeVO2_enrichGO("t12", "negative", "Up", H12vsPRE)
H24correlationnegUp <- timeVO2_enrichGO("t24", "negative", "Up", H24vsPRE)
H48correlationnegUp <- timeVO2_enrichGO("t48", "negative", "Up", H48vsPRE)
H3correlationposDown <- timeVO2_enrichGO("t3", "positive", "Down", H3vsPRE)
H6correlationposDown <- timeVO2_enrichGO("t6", "positive", "Down", H6vsPRE)
H9correlationposDown <- timeVO2_enrichGO("t9", "positive", "Down", H9vsPRE)
H12correlationposDown <- timeVO2_enrichGO("t12", "positive", "Down", H12vsPRE)
H24correlationposDown <- timeVO2_enrichGO("t24", "positive", "Down", H24vsPRE)
H48correlationposDown <- timeVO2_enrichGO("t48", "positive", "Down", H48vsPRE)
H3correlationnegDown <- timeVO2_enrichGO("t3", "negative", "Down", H3vsPRE)
H6correlationnegDown <- timeVO2_enrichGO("t6", "negative", "Down", H6vsPRE)
H9correlationnegDown <- timeVO2_enrichGO("t9", "negative", "Down", H9vsPRE)
H12correlationnegDown <- timeVO2_enrichGO("t12", "negative", "Down", H12vsPRE)
H24correlationnegDown <- timeVO2_enrichGO("t24", "negative", "Down", H24vsPRE)
H48correlationnegDown <- timeVO2_enrichGO("t48", "negative", "Down", H48vsPRE)

timecorrelation <- rbind(H3correlationposUp, H6correlationposUp, H9correlationposUp, H12correlationposUp, H24correlationposUp, H48correlationposUp,H3correlationnegUp, H6correlationnegUp, H9correlationnegUp, H12correlationnegUp, H24correlationnegUp, H48correlationnegUp, H3correlationposDown, H6correlationposDown, H9correlationposDown, H12correlationposDown, H24correlationposDown, H48correlationposDown,H3correlationnegDown, H6correlationnegDown, H9correlationnegDown, H12correlationnegDown, H24correlationnegDown, H48correlationnegDown)

timecorrelation <- timecorrelation %>%
  mutate(Description = if_else(
    str_starts(Description, "rRNA") | str_starts(Description, "ncRNA") | str_starts(Description, "mRNA"),
    Description, str_to_title(Description)))

timecorrelation <- timecorrelation %>%
 mutate(
    effect = case_when(
      Direction == "positive" & Regulation == "Down" ~ "Low",
      Direction == "positive" & Regulation == "Up" ~ "High",
      Direction == "negative" & Regulation == "Down" ~ "High",
      Direction == "negative" & Regulation == "Up" ~ "Low",
      TRUE ~ NA_character_))

timecorrelation$GeneRatio <- sapply(strsplit(timecorrelation$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

timecorrelation <- timecorrelation %>%
  mutate(Index = recode(Index,
                        "t3" = "3H",
                        "t6" = "6H",
                        "t9" = "9H",
                        "t12" = "12H",
                        "t24" = "24H",
                        "t48" = "48H"))

timecorrelation$Index <- factor(timecorrelation$Index, levels = c("3H", "6H", "9H", "12H", "24H", "48H"))

timecorrelation$effect <- factor(timecorrelation$effect , levels = c("Low", "High"))

timecorrelation$Regulation <- factor(timecorrelation$Regulation , levels = c("Up", "Down"))

time_background <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      "3H" = "#984EA3",
      "12H" = "#A65628",
      "24H" = "#F781BF",
      "48H" = "#999999",
      "Up" = "white",
      "Down" = "white",
      "Up" = "white",
      "Down" = "white",
      "Up" = "white",
      "Down" = "white",
      "Up" = "white",
      "Down" = "white")))

timecorrelation$Description  <- factor(timecorrelation$Description, levels = unique(timecorrelation$Description ))

ggplot(timecorrelation, aes(x = effect, y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_y_discrete(limits = rev(levels(timecorrelation$Description))) +
  scale_size_continuous(name = "Number of genes", range = c(1, 5)) +
  scale_color_gradientn(name = "Adjusted p-value", colours = c("red", "blue"),
                        limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01),
                        na.value = "grey50") +
  theme_minimal() +
  labs(y = NULL,
       x = "Cardiorespiratory fitness for which a greater magnitude of expression was observed") +
  ggtitle(bquote("Correlation between gene expression and baseline" ~ VO[2*"peak"])) +
  theme(axis.text.y = element_text(size = 9), legend.position = "right",
        axis.title.x = element_text(size = 10),
        panel.border = element_rect(color = "black", fill = NA),
        strip.background = element_rect(fill = "grey80", color = "grey50"),
        strip.text = element_text(face = "bold", size = 10),
        legend.text = element_text(size = 8),
        axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        axis.ticks.length = unit(0.1, "cm"),
        plot.margin=unit(c(0,0,0,0.5),"cm")) + 
  facet_nested(~ Index * Regulation, scales = "free_y",
               switch = "y", space = "free_y",
               strip = time_background) +
  guides(size = guide_legend(legend.position = "top"),
         color = guide_colorbar(title.position = "top"))
```

#enrichment of correlation between VO2 and genes using gene features
```{r}
featureVO2_enrichGO <- function(feature, directionVO2) {
  
  correlatedVO2 <- VO2correlatedfilter %>%
  dplyr::filter(index == feature & direction == directionVO2 & regulation == "feature") %>%
  dplyr::pull(gene)

  correlatedVO2 <- enrichGO(correlatedVO2, OrgDb = "org.Hs.eg.db",
                                 universe=rownames(rna_comp), keyType = "SYMBOL",
                                 ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.2,
                                 pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

  correlatedVO2 <- as.data.frame(correlatedVO2@result) %>% arrange(p.adjust) %>%
  dplyr::filter(p.adjust < 0.05) %>% head(5) %>%
  mutate(Index = feature, Direction = directionVO2)
  
  return(correlatedVO2)
}

peakfcupcorrelationpos<- featureVO2_enrichGO("peakfcup", "positive")
peakfcupcorrelationneg <- featureVO2_enrichGO("peakfcup", "negative")
peakfcdowncorrelationpos<- featureVO2_enrichGO("peakfcdown", "positive")
peakfcdowncorrelationneg <- featureVO2_enrichGO("peakfcdown", "negative")
avgfccorrelationpos<- featureVO2_enrichGO("avgfc", "positive")
avgfccorrelationneg <- featureVO2_enrichGO("avgfc", "negative")
genestdevcorrelationpos<- featureVO2_enrichGO("genestdev", "positive")
genestdevcorrelationneg <- featureVO2_enrichGO("genestdev", "negative")
auccorrelationpos<- featureVO2_enrichGO("auc", "positive")
auccorrelationneg <- featureVO2_enrichGO("auc", "negative")

featurecorrelation <- rbind(peakfcupcorrelationpos, peakfcdowncorrelationpos, avgfccorrelationpos, genestdevcorrelationpos, auccorrelationpos, peakfcupcorrelationneg, peakfcdowncorrelationneg, avgfccorrelationneg, genestdevcorrelationneg, auccorrelationneg)

featurecorrelation <- featurecorrelation %>%
  mutate(Description = if_else(str_starts(Description, "rRNA") | str_starts(Description, "ncRNA") | str_starts(Description, "mRNA"), Description, str_to_title(Description)))

featurecorrelation$GeneRatio <- sapply(strsplit(featurecorrelation$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

featurecorrelation <- featurecorrelation %>%
  mutate(Index = recode(Index,
                        "peakfcup" = "Peak FC up",
                        "peakfcdown" = "Peak FC down",
                        "avgfc" = "Average FC",
                        "genestdev" = "Standard deviation",
                        "auc" = "AUC"))

featurecorrelation$Index <- factor(featurecorrelation$Index, levels = c("Peak FC up", "Peak FC down", "Average FC", "Standard deviation", "AUC"))

featurecorrelation$Direction <- factor(featurecorrelation$Direction , levels = c("negative", "positive"))

feature_background <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      "Peak FC up" = "#8DD3C7",
      "Average FC" = "#BEBADA",
      "Standard deviation" = "#FB8072",
      "AUC" = "#80B1D3")))

featurecorrelation$Description[featurecorrelation$Description == "maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)"] <- "maturation of SSU-rRNA from tricistronic rRNA transcript"

ggplot(featurecorrelation, aes(x = Direction, y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_x_discrete(labels = c("negative" = "Low", "positive" = "High")) +
  scale_y_discrete(limits = rev(levels(featurecorrelation$Description))) +
  scale_size_continuous(name = "Number of genes", range = c(0.5, 5)) +
  scale_color_gradientn(name = "Adjusted p-value", colours = c("red", "blue"),
                        limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01),
                        na.value = "grey50") +
  theme_minimal() +
  labs(y = NULL, x = "Cardiorespiratory fitness for which a greater magnitude of expression was observed") +
ggtitle(bquote("Correlation between gene expression and baseline" ~ VO[2*"peak"])) +
  theme(axis.text.y = element_text(size = 9),
        axis.title.x = element_text(size = 10),
        legend.position = "right",
        panel.border = element_rect(color = "black", fill = NA),
        strip.background = element_rect(fill = "grey80", color = "grey50"),
        strip.text = element_text(face = "bold", size = 10),
        legend.text = element_text(size = 8),
        axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        axis.ticks.length = unit(0.1, "cm")) + 
  facet_nested(~ Index, scales = "free_y", switch = "y", space = "free_y",
               strip = feature_background) +
  guides(size = guide_legend(legend.position = "top"), color = guide_colorbar(title.position = "top"))
```

#piechart of how many mito genes are DE of those within mitocarta3.0 (3.7A)
```{r}
mitocarta <- read.xlsx("data/MitoCarta30.xlsx")
Gomit <- mitocarta
notdetectedmito <- setdiff(mitocarta$Symbol,rownames(rna_comp))

mitoDE <- merge(rna_comp, Gomit, by = "Symbol", "row.names", keep.rownames = TRUE)
rownames(mitoDE) <- mitoDE$Row.names

mitototal <- as.numeric(nrow(mitocarta))
mitoDEgene <- as.numeric(nrow(mitoDE))
notDEmito <- mitototal - mitoDEgene

Mitopiechart <- data.frame(
  category = c("Mitochondrial genes DE in response to exercise", "Mitochondrial genes not DE in response to exercise"),
  value = c(mitoDEgene, notDEmito))

category_colors <- c("#d65c5c", "grey")

ggplot(Mitopiechart, aes(x = "", y = value, fill = category)) +
  geom_bar(width = 1, stat = "identity") +
  scale_fill_manual(values = category_colors) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = NULL) +
  geom_text(aes(label = paste0(value, "")),
            position = position_stack(vjust = 0.5),
            size = 10)+
  theme(legend.text = element_text(size = 12),
  plot.title = element_text(size = 20))
```

#no. mito DEGs at each time point and average expression, with no. previously ided genes indicated (3.7B)
```{r}
#comparing to old matrix
TotalDEsubset <- function(contrast, subset) {
  rownames_totalDE <- rownames(contrast)
  filtered_rows <- rownames_totalDE[rownames_totalDE %in% subset]
  return(contrast[filtered_rows, , drop = FALSE])}

# filter total DEGs by MitoCarta3.0
MitDE1 <- TotalDEsubset(MIDvsPRE, Gomit$Symbol)
MitDE2 <- TotalDEsubset(POSTvsPRE, Gomit$Symbol)
MitDE3 <- TotalDEsubset(H3vsPRE, Gomit$Symbol)
MitDE4 <- TotalDEsubset(H6vsPRE, Gomit$Symbol)
MitDE5 <- TotalDEsubset(H9vsPRE, Gomit$Symbol)
MitDE6 <- TotalDEsubset(H12vsPRE, Gomit$Symbol)
MitDE7 <- TotalDEsubset(H24vsPRE, Gomit$Symbol)
MitDE8 <- TotalDEsubset(H48vsPRE, Gomit$Symbol)

MitoDE1 <- process_DE_genes(MitDE1, "MID", previous_genes)
MitoDE2 <- process_DE_genes(MitDE2, "POST", previous_genes)
MitoDE3 <- process_DE_genes(MitDE3, "3H", previous_genes)
MitoDE4 <- process_DE_genes(MitDE4, "6H", previous_genes)
MitoDE5 <- process_DE_genes(MitDE5, "9H", previous_genes)
MitoDE6 <- process_DE_genes(MitDE6, "12H", previous_genes)
MitoDE7 <- process_DE_genes(MitDE7, "24H", previous_genes)
MitoDE8 <- process_DE_genes(MitDE8, "48H", previous_genes)

MitDE <- rbind (MitoDE1, MitoDE2, MitoDE3, MitoDE4, MitoDE5, MitoDE6, MitoDE7, MitoDE8)

MitDE$Time <- as.character(MitDE$Time)
MitDE$Time <- factor(MitDE$Time, levels = unique(MitDE$Time))

ggplot(data = MitDE, aes(x = Time, y = Genes, fill = Direction, pattern = Previous)) + 
   geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
  labs(x = "Time", y = "No. DEGs relative to PRE", fill = "Number of DEGs",
       pattern =str_wrap("Previously identified as exercise-induced DEG", width = 25)) +
  scale_fill_manual(values=c("Up" = "#ed3434", "Down" = "#b03333"),
                    labels = c("Up" = "Upregulated", "Down" = "Downregulated")) +
      scale_pattern_manual(values = c("Never" = "none", "Previously identified" = "stripe"),
                           labels = c("Previously identified" = "Yes", "Never" = "No")) +
  geom_point(data = MitDE, aes(x=Time, y=meanlogFC * 500, shape = Direction), size = 4) +
  geom_errorbar(data = MitDE, aes(ymax=(meanlogFC + sdlogFC)*500, ymin=(meanlogFC - sdlogFC)*500),
                width = 0.5, size = 0.75) + 
  scale_shape_manual(name = "Average DEG fold change", values = c(Up=15, Down = 18),
                   labels = c("Up" = "Upregulated genes", "Down" = "Downregulated genes")) +
  scale_y_continuous(name = expression("        No. DEGs relative to PRE\n \n Downregulated              Upregulated"), 
                    sec.axis = sec_axis(~.*0.002,
                      name = bquote("Average" ~~ log[2] ~ "expression of DEGs relative to PRE"),
                      breaks = c(-1, -0.5, 0, 0.5, 1)),
                    labels = function(x) abs(x),
                    limits = c(-600, 600),
                    breaks=c(-600, -400, -200, 0, 200, 400, 600)) +
  theme_set(theme_bw()) +
  theme(axis.title.y = element_text(margin = ggplot2::margin(r = 5, l = 20)),
        axis.title.y.right = element_text(margin = ggplot2::margin(l = 5, r =5))) +
  ggtitle("Exercise-induced mitochondrial DEGs (Mitocarta 3.0)") +
  guides(fill = "none",
         pattern = guide_legend(reverse = TRUE, override.aes = list(fill = "white")),
         shape = guide_legend(reverse = TRUE))
```

#Soft clustering using c-means with mfuzz package (3.7C)
```{r}
#Soft clustering of mito DEGs using fuzzy c-means
mitoDEsoft <- mitoDE[,2:9]
colnames(mitoDEsoft) <- c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")

mitoDEsoft <- as.matrix(mitoDEsoft)
tmp_expr <- new('ExpressionSet', exprs = mitoDEsoft)

#estimation of 'fuzzifier' constant
m1 <- mestimate(tmp_expr)

#estimation and visualisation of number of clusters (no super clean elbow)
ggplot() +
  geom_text(x = 0.5, y = 0.5,
            aes(label = "Minimum centroid distance"),
            size = 3, angle = 90) +
  coord_fixed(ylim = c(0, 1), xlim = c(0, 1))

Dmin(tmp_expr,m1,crange=seq(2,20,2),repeats=1,visu=TRUE)

#Soft clustering of data
fuzzcluster <- mfuzz(tmp_expr, centers = 6, m = m1 )

#Plot of soft clustering of data
mfuzz.plot2(tmp_expr, cl = fuzzcluster, mfrow = c(2, 4), colo="fancy",
            min.mem = 0.3, time.labels = colnames(mitoDEsoft),
            xlab="Time", ylab =(Log[2]*FC~ "relative to PRE"),
            ax.col="black",bg = "white",col.axis="black",col.lab="black",
            col.main="black",col.sub="black",col="black",
            centre.col="green",centre.lwd=3, centre=TRUE, x11=FALSE)

mfuzzColorBar(col="fancy", horizontal=FALSE, main="Gene membership value", cex.main=1.5)

#PCA of each cluster to determine overlap (as it's soft clustering) -there is threshold of membership of each genes - overlap represented by lines between clusters
overlapfc <- Mfuzz::overlap(fuzzcluster)
rownames(overlapfc) <- c("Cluster 1", "Cluster 4", "Cluster 5", "Cluster 2", "Cluster 6", "Cluster 3")
overlapfc <- overlapfc[c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6"),]

colnames(overlapfc) <- c("Cluster 1", "Cluster 4", "Cluster 5", "Cluster 2", "Cluster 6", "Cluster 3")
overlapfc <- overlapfc[, c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6")]

my_colors <- colorRamp2(c(0, 1), c("white", "darkred"))

heatmap_obj <- Heatmap(overlapfc, 
                       cluster_rows = FALSE,
                       cluster_columns = FALSE,
                       show_row_names = TRUE,
                       show_column_names = TRUE,
                       row_names_side = "left",
                       column_names_side = "top",
                       column_names_rot = 45,
                       col = my_colors,
                      row_names_gp = gpar(fontsize = 10),
                      column_names_gp = gpar(fontsize = 10),
                      column_title = "Genes within mitochondrial cluster",
                      row_title = "Expression pattern",
                      heatmap_legend_param = list(
    title = "Cluster overlap", 
    legend_width = unit(2.5, "cm"),
    legend_direction = "horizontal"))


draw(heatmap_obj, heatmap_legend_side = "bottom")

#Determination of number of genes within cluster (only considers a genes highest membership - hard cluster for sake of identification lol)
fuzzcluster$size
clustersize <- fuzzcluster$size

cluster_count <- data.frame(
  Total_Count = clustersize,
  Cluster = c("Cluster 1", "Cluster 4", "Cluster 5", "Cluster 2", "Cluster 6", "Cluster 3"))

#generation of gene sets within all clusters
clusterprofile <- acore(tmp_expr,fuzzcluster, min.acore=0.1)

#selection of genes within a specific cluster
cluster1 <- clusterprofile[[1]]
#openxlsx::write.xlsx(cluster1,"cluster1.xlsx", rowNames = TRUE)
cluster1 <- openxlsx::read.xlsx("data/cluster1.xlsx")
genes_to_test1 <- cluster1$NAME

cluster2 <- clusterprofile[[4]]
#openxlsx::write.xlsx(cluster2,"cluster2.xlsx", rowNames = TRUE)
cluster2 <- openxlsx::read.xlsx("data/cluster2.xlsx")
genes_to_test2 <- cluster2$NAME

cluster3 <- clusterprofile[[6]]
#openxlsx::write.xlsx(cluster3,"cluster3.xlsx", rowNames = TRUE)
cluster3 <- openxlsx::read.xlsx("data/cluster3.xlsx")
genes_to_test3 <- cluster3$NAME

cluster4 <- clusterprofile[[2]]
#openxlsx::write.xlsx(cluster4,"cluster4.xlsx", rowNames = TRUE)
cluster4 <- openxlsx::read.xlsx("data/cluster4.xlsx")
genes_to_test4 <- cluster4$NAME

cluster5 <- clusterprofile[[3]]
#openxlsx::write.xlsx(cluster5,"cluster5.xlsx", rowNames = TRUE)
cluster5 <- openxlsx::read.xlsx("data/cluster5.xlsx")
genes_to_test5 <- cluster5$NAME

cluster6 <- clusterprofile[[5]]
#openxlsx::write.xlsx(cluster6,"cluster6.xlsx", rowNames = TRUE)
cluster6 <- openxlsx::read.xlsx("data/cluster6.xlsx")
genes_to_test6 <- cluster6$NAME

membership <- fuzzcluster$membership
colnames(membership) <- c("one", "two", "three", "four", "five", "six")

create_cluster_plot <- function(gene_cluster_list, membership_column, cluster_name) {
  cluster_data <- rna_comp[rownames(rna_comp) %in% gene_cluster_list, ]
  cluster_data <- cluster_data[, 1:8]
  cluster_data <- merge(cluster_data, membership, by.x = 0, by.y = 0)
  colnames(cluster_data) <- c("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", "one", "two", "three", "four", "five", "six")
  
cluster_data <- cluster_data %>% dplyr::select("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", membership_column)

cluster_plot_data <- gather(cluster_data, key = "Time", value = "LogFC", -Gene, -{{membership_column}})
  cluster_plot_data$Time <- as.character(cluster_plot_data$Time)
  cluster_plot_data$Time <- factor(cluster_plot_data$Time, levels = unique(cluster_plot_data$Time))
  cluster_plot_data <- na.omit(cluster_plot_data)
  
cluster_plot <- ggplot(cluster_plot_data, aes(y=LogFC, x=Time, group=Gene, color=!!sym(membership_column))) + 
    labs(x = "Time", y = bquote(Log[2] * FC ~ "relative to PRE"))  +
    geom_line(data = cluster_plot_data, size = 0.3) +
    scale_color_gradientn(name = "Cluster membership",
                          colours = c("yellow", "navy"),
                          limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4),
                          guide = "none") + 
    stat_summary(aes(group = 1), geom = "line", fun = "mean", colour = "green", size = 1.5) +
    ggtitle(sprintf("%s (n = %d)", cluster_name, length(gene_cluster_list))) +
    theme_set(theme_bw())
  
  return(cluster_plot)
}

cluster1_plot <- create_cluster_plot(genes_to_test1, "one", "Cluster 1")
cluster2_plot <- create_cluster_plot(genes_to_test2, "four", "Cluster 2")
cluster3_plot <- create_cluster_plot(genes_to_test3, "six", "Cluster 3")
cluster4_plot <- create_cluster_plot(genes_to_test4, "two", "Cluster 4")
cluster5_plot <- create_cluster_plot(genes_to_test5, "three", "Cluster 5")
cluster6_plot <- create_cluster_plot(genes_to_test6, "five", "Cluster 6")

membership_legend <- ggplot(data.frame(value = c(0, 1)),
                            aes(x = 1, y = value, color = value)) +
                            geom_point(size = 0) +
                            scale_color_gradientn(name = "Cluster membership", colours = c("yellow", "navy"), limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4)) +
                            theme_void() +
                            theme(legend.position = "left")

grid.arrange(cluster1_plot, cluster2_plot, cluster3_plot, cluster4_plot, cluster5_plot, cluster6_plot, ncol=2, widths = c(3, 3))
```

#creating an enrichment plot using the 2nd order Mitochondrial pathways in Mitocarta3.0 (3.7F)
```{r}
#creating the mitochondrial pathway enrichment plot for the different clusters
mitocartaadj <- read.xlsx("data/mitocarta30adj.xlsx")
colnames(mitocartaadj) <- c("term", "gene", "hier")
mitocartaadj <- mitocartaadj %>%
  mutate(gene = strsplit(as.character(gene), ", ")) %>%
  unnest(cols = gene)

allmitoDEGs <- intersect(mitocarta$Symbol, rownames(rna_comp))

mitoclusterenrichment <- function(genes_to_test_list) {
  result_list <- lapply(seq_along(genes_to_test_list), function(i) {
  enrich_result <- enricher(genes_to_test_list[[i]],
                            universe = allmitoDEGs,
                            pAdjustMethod = "BH", minGSSize = 1,
                            gson = NULL, TERM2GENE = mitocartaadj)
  df_result <- as.data.frame(enrich_result@result) %>%
      arrange(pvalue) %>%
      mutate(Condition = paste("Cluster", i))
  
  return(df_result)})
  
  combined_results <- do.call(rbind, result_list)
  
  return(combined_results)}

combinedMCenrichment <- mitoclusterenrichment(list(genes_to_test1, genes_to_test2, genes_to_test3, genes_to_test4, genes_to_test5, genes_to_test6))

combinedMCenrichment$GeneRatio <- sapply(strsplit(combinedMCenrichment$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

condition_order2 <- c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6")

combinedMCenrichment$Condition <- factor(combinedMCenrichment$Condition, levels = condition_order2)

combinedMCenrichment$Description <- fct_rev(combinedMCenrichment$Description)

ggplot(combinedMCenrichment, aes(x = Condition, y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_size_continuous(name = "Genes", range = c(1.5, 6)) +
  scale_color_gradientn(name = "Adjusted p-value", colours = c("red", "blue"),
                        limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01),
                        na.value = "black") +
  theme_minimal() +
  labs(y=NULL) +
  ggtitle("Mitochondrial gene cluster enrichment (MitoCarta 3.0)") +
  theme(panel.border = element_rect(color = "black", fill = NA),
        strip.text.y = element_text(angle = 90, size = 8, face="bold"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 8),
        axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        axis.ticks.length = unit(0.1, "cm"),
        strip.background = element_rect(fill = "white"),
        panel.spacing = unit(0, "cm"),
        strip.placement = "outside",
        plot.margin = ggplot2::margin(0.3, 0.3, 0.3, 0.3, "cm")) +
  guides(size = guide_legend(), color = guide_colorbar(title.position = "top"))
```

#looking at baseline sex-specific differences
```{r}
designbaseline <- model.matrix(~0 + group:sex, y$samples)
colnames(designbaseline) <- gsub("group", "", colnames(designbaseline))
colnames(designbaseline) <- gsub(":sex", "", colnames(designbaseline))

contr.matrixbaseline <- makeContrasts(
  "MIDvsPRE" = PREM-PREF,
  levels = colnames(designbaseline))
contr.matrixbaseline

vbaseline <- voom(y, designbaseline, plot=TRUE)

corbaseline <- duplicateCorrelation(counts, designbaseline, block=sampleid)
corbaseline$consensus.correlation
fit2baseline <- lmFit(vbaseline, design=designbaseline, 
  block=sampleid, correlation=corbaseline$consensus.correlation)

fit2baseline <- lmFit(vbaseline, designbaseline)
fit2baseline <- contrasts.fit(fit2baseline, contrasts=contr.matrixbaseline)
efit2baseline <- eBayes(fit2baseline)

summary(decideTests(efit2baseline, adjust.method = "BH", p.value=0.05))

rna_compbaseline <- topTable(efit2baseline, n=Inf, adjust.method = "BH")
rna_compbaselineALL <- topTable(efit2baseline, n=Inf, adjust.method = "BH")
rna_compbaseline <- subset(rna_compbaseline, rna_compbaseline$adj.P.Val<0.05)
openxlsx::write.xlsx(rna_compbaseline,"rna_compsexbaselinegenes.xlsx", rowNames = TRUE)
```

#unbatched MDS of samples for each time point (3.9A)
```{r}
mds_dataunbatched <- plotMDS.DGEList(y)
mds_dataunbatched  <- data.frame(x=mds_dataunbatched$x,y=mds_dataunbatched$y,sampledata)

timecolours <-  brewer.pal(9, "Set1")
legend_label <- c("PRE" = "PRE", "MID" = "MID", "POST"="POST", "threeH"="3H", "sixH"="6H", "nineH"="9H", "twelveH"="12H", "twentyfourH"="24H", "fourtyeightH"="48H")

ggplot(mds_dataunbatched,aes(x=x,y=y)) +
      geom_point(aes(color=as.factor(time), shape=as.factor(sex)),size=4) +
      labs(x = "Dimension 1", y = "Dimension 2", colour = "Time point", shape = "Sex") +
      scale_x_continuous(breaks = pretty(mds_dataunbatched$x, n=14)) +
      scale_y_continuous(breaks = pretty(mds_dataunbatched$y, n=14)) +
      scale_colour_manual(values = timecolours,legend_label,
                          labels = legend_label,
                          limits=c("PRE", "MID", "POST", "threeH", "sixH", "nineH", "twelveH", "twentyfourH", "fourtyeightH")) +
      theme(axis.title.x = element_text(vjust = 0, size = 20),
      axis.title.y = element_text(vjust = 0, size = 20)) +
      scale_shape_manual(values=c(F=16,M=17),labels=c("F"="Women", "M"="Men")) + 
      theme(legend.txt = element_text(size =20)) +
      ggtitle("Unbatched RNA-seq Time Course MDS") +
      theme(plot.title = element_text(size = 100)) +
      theme_bw()
```

#plot of baseline sex-specific differences from each chromosome (3.9B)
```{r}
malebaseline <- dplyr::filter(rna_compbaseline, rna_compbaseline$logFC > 0)
openxlsx::write.xlsx(malebaseline,"malebaseline.xlsx", rowNames = TRUE)
femalebaseline <- dplyr::filter(rna_compbaseline, rna_compbaseline$logFC < 0)
openxlsx::write.xlsx(femalebaseline,"femalebaseline.xlsx", rowNames = TRUE)

listofchromosomes <- c('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y')

#read in list of upregulated genes in men and their corresponding chromosome from biomart
malebaselinechromosomes <- read_tsv("data/malebaselinemart_export.txt")
malebaselineindex <- malebaselinechromosomes$`Chromosome/scaffold name` %in% listofchromosomes
malebaseline <- malebaselinechromosomes[malebaselineindex, ] %>% mutate(Sex = "Men")

#read in list of upregulated genes in women and their corresponding chromosome from biomart
femalebaselinechromosomes <- read_tsv("data/femalebaselinemart_export.txt")
femalebaselineindex <- femalebaselinechromosomes$`Chromosome/scaffold name` %in% listofchromosomes
femalebaseline <- femalebaselinechromosomes[femalebaselineindex, ] %>% mutate(Sex = "Women")

#read in list of all genes in men and women and their corresponding chromosome from biomart
allbaselinechromosomes <- read_tsv("data/allbaselinemart_export.txt")
allbaselineindex <- allbaselinechromosomes$`Chromosome/scaffold name` %in% listofchromosomes
allbaseline <- allbaselinechromosomes[allbaselineindex, ] %>% mutate(Sex = "All")

Baselinechromosome <- rbind(malebaseline, femalebaseline, allbaseline)
colnames(Baselinechromosome) <- c("Chromosome", "Gene", "Sex")

chromosomes <- character()
men_percentages <- numeric()
women_percentages <- numeric()

# Calculate the counts and ratios for each unique chromosome value for Men and Women
for (value in listofchromosomes) {
  # Calculate for Men
  count_men <- sum(Baselinechromosome$Chromosome == value & Baselinechromosome$Sex == 'Men')
  count_all_men <- sum(Baselinechromosome$Chromosome == value & Baselinechromosome$Sex == 'All')
  ratio_men <- if (count_all_men != 0) (count_men / count_all_men) * 100 else NA
  
  # Calculate for Women
  count_women <- sum(Baselinechromosome$Chromosome == value & Baselinechromosome$Sex == 'Women')
  count_all_women <- sum(Baselinechromosome$Chromosome == value & Baselinechromosome$Sex == 'All')
  ratio_women <- if (count_all_women != 0) (count_women / count_all_women) * 100 else NA
  
  # Store the results
  chromosomes <- c(chromosomes, value)
  men_percentages <- c(men_percentages, ratio_men)
  women_percentages <- c(women_percentages, ratio_women)
}

Baselinechromosome <- data.frame(
  Chromosome = chromosomes,
  Men_Percentage = men_percentages,
  Women_Percentage = women_percentages)

Baselinechromosome <- pivot_longer(
  Baselinechromosome,
  cols = c(Men_Percentage, Women_Percentage),
  names_to = "Sex",
  values_to = "Percentage"
)

Sexcolours <- c("Men_Percentage" = "#02e1f5", "Women_Percentage" = "#f02bd9")
Baselinechromosome$Chromosome <- as.character(Baselinechromosome$Chromosome)
Baselinechromosome$Chromosome <- factor(Baselinechromosome$Chromosome, levels = listofchromosomes)

#create a plot of what chromosome the genes that are different between sexes at baseline are from
ggplot(Baselinechromosome, aes(x = Chromosome, fill = Sex, y = Percentage)) +
  geom_col() +
  labs(x= "Chromosome", y= "Percentage of sex-specific DEGs (%)", fill="Sex") + 
  scale_fill_manual(values = Sexcolours,
                    labels = c("Men_Percentage" = "Men",
                               "Women_Percentage" = "Women")) + 
  guides(fill = guide_legend(reverse = TRUE)) +
  theme_bw() +
  theme(axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14),
        plot.title = element_text(size = 16)) +
  scale_y_continuous(expand = expansion(mult = c(0, 0)), limits= c(0, 100),breaks = seq(0, 100, by = 20)) +
  ggtitle("Percentage of genes expressed from each chromosome with sex-specific differential expression at baseline")
```

#comparing baseline sex difs in our study to other studies (3.9C)
```{r}
chapman2020control <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/ChapmanMCvsFC2020.xlsx")
chapman2020controlgenes <- unique(chapman2020control$gene_name)

chapman2020trained <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/ChapmanMEvsFE2020.xlsx")
chapman2020trainedgenes <- unique(chapman2020trained$gene_name)

maher2009 <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/Maher2009.xlsx")
maher2009genes <- na.omit(maher2009[,c("Acronym")])

#only lists top 500 so ignore
oliva2020 <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/Olivia2020.xlsx")
oliva2020genes <- na.omit(oliva2020[,c("HUGO_gene_id")])

welle2008 <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/Welle2008.xlsx")
welle2008 <- welle2008 %>% dplyr::filter(welle2008$Symbol != "---")
welle2008$adjustedpval <- p.adjust(welle2008$P.sex, method = "BH")
welle2008 <- welle2008[welle2008$adjustedpval < 0.05, ]
welle2008genes <- unique(welle2008$Symbol)

gershoni2017 <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/Gershoni2017.xlsx")
gershoni2017 <- subset(gershoni2017,select = c("Gene", "Muscle-Skeletal"))
gershoni2017 <- gershoni2017 %>% dplyr::filter(gershoni2017$`Muscle-Skeletal` !=0)
gershoni2017 <- gershoni2017 %>% mutate(first_word = word(Gene, 1, sep = "_")) %>% mutate(Gene = first_word) %>% dplyr::select(-first_word)
gershoni2017genes <- unique(gershoni2017$Gene)

lopesramos2020 <- openxlsx::read.xlsx("data/Previousstudies/sexbaseline/Lopes-Ramos2020.xlsx")
lopesramos2020genes <- unique(lopesramos2020$ID)

taylor2024genes <- rownames(rna_compbaseline)
  
previoussexstudies <- list(
 "Lopes-Ramos et al. 2020" =  lopesramos2020genes,
  "Gershoni et al. 2017" = gershoni2017genes,
 "Maher et al. 2009" = maher2009genes,
 "Oliva et al. 2020 (Top 500 genes by FDR)" = oliva2020genes,
 "Current study" = taylor2024genes,
 "Chapman et al. 2020 (Control)" = chapman2020controlgenes,
 "Chapman et al. 2020 (Trained)" = chapman2020trainedgenes
)

wholelist <- unlist(previoussexstudies)
wholelist <-as.data.frame(wholelist)
openxlsx::write.xlsx(wholelist,"sexbaselinegenes.xlsx", rowNames = TRUE)
wholelist$study <- rownames(wholelist)
beep <- merge(wholelist, mitocarta,  by.y ="Symbol",  by.x =  "wholelist")

boop <- intersect(wholelist$wholelist, mitocarta$Symbol)

#create the upset plot to show overlap and differences
UpSetR::upset(fromList(previoussexstudies), nsets=6, order.by = "freq", nintersects=25,  mb.ratio = c(0.6, 0.4),
      number.angles = 15, 
      text.scale = 1.2, 
      point.size = 2.5, 
      line.size = 1,
      mainbar.y.label = "Overlapping genes",
      sets.x.label = "Number of sex-specific genes")

combinations <- list(c("lopesramos2020genes", "gershoni2017genes", "maher2009genes", "welle2008genes", "taylor2024genes","chapman2020trainedgenes", "chapman2020controlgenes"))

sexcombination1 <- intersect(chapman2020controlgenes, rownames(rna_compbaseline))
sexcombination2 <- intersect(lopesramos2020genes, rownames(rna_compbaseline))
sexcombination3 <- intersect(gershoni2017genes, rownames(rna_compbaseline))
sexcombination4 <- intersect(chapman2020trainedgenes, rownames(rna_compbaseline))
sexcombination5 <- intersect(maher2009genes, rownames(rna_compbaseline))
sexcombination6 <- intersect(oliva2020genes, rownames(rna_compbaseline))

sexcombination <- unique(c(sexcombination1, sexcombination2, sexcombination3, sexcombination4,
                           sexcombination5, sexcombination6))
sexcombination <- as.data.frame(sexcombination)

openxlsx::write.xlsx(sexcombination, "sexcombos.xlsx")

sexcombinations <- read_tsv("data/sexbaselinecombos.txt")
sexcombinations <- sexcombinations[!duplicated(sexcombinations$`HGNC symbol`), ]

replacement <- gsub("^HSCHR([0-9]+).*$", "\\1", sexcombinations$`Chromosome/scaffold name`)

sexcombinations$`Chromosome/scaffold name` <- ifelse(grepl("^HSCHR", 
                                            sexcombinations$`Chromosome/scaffold name`),
                                          replacement, sexcombinations$`Chromosome/scaffold name`)

rows_to_replace <- sexcombinations$`HGNC symbol` == "GCNT2"
sexcombinations$`Chromosome/scaffold name`[rows_to_replace] <- "6"

rows_to_replace <- sexcombinations$`HGNC symbol` == "XYLT1"
sexcombinations$`Chromosome/scaffold name`[rows_to_replace] <- "16"

sexcombinations <- na.omit(sexcombinations)

sexcombinations$ChromosomeClass <- ifelse(sexcombinations$`Chromosome/scaffold name` %in%
                                            c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21"), "Autosomal", 
                              ifelse(sexcombinations$`Chromosome/scaffold name` == "X", "X", "Y"))

chromosome_counts <- table(sexcombinations$ChromosomeClass)

chromosome_counts_df <- as.data.frame(chromosome_counts)
names(chromosome_counts_df) <- c("ChromosomeClass", "Count")

custom_colors <- c("Autosomal" = "grey", "X" = "green", "Y" = "purple")

ggplot(chromosome_counts_df, aes(x = "", y = Count, fill = ChromosomeClass)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start=0) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 8) +
  theme_void() +
  scale_fill_manual(values = custom_colors) + 
  labs(fill = str_wrap("Genes with sex-specific differences at baseline shared between this study and other studies", width=25))

sexcombinationsautosomal <- subset(sexcombinations, grepl("Autosomal", ChromosomeClass))

autosomalenrichgo <- enrichGO(sexcombinationsautosomal$`HGNC symbol`,
                              OrgDb = "org.Hs.eg.db",
                              universe=rownames(rna_compbaselineALL),
                              keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05,
                              qvalueCutoff = 0.2, pAdjustMethod = "BH",
                              readable = FALSE, pool = FALSE)

#shared in 2 or more studies?
all_genes <- unlist(previoussexstudies)

gene_counts <- table(all_genes)

common_genes <- names(gene_counts[gene_counts >= 2])

common_genes <- as.data.frame(common_genes)

#openxlsx::write.xlsx(common_genes,"sharedsexbaselineover1study.xlsx", rowNames = TRUE)

sexcombinations2 <- read_tsv("data/allsharedsexbaseline.txt")
sexcombinations2 <- sexcombinations2[!duplicated(sexcombinations2$`Gene name`), ]

replacement <- gsub("^HSCHR([0-9]+).*$", "\\1", sexcombinations2$`Chromosome/scaffold name`)

sexcombinations2$`Chromosome/scaffold name` <- ifelse(grepl("^HSCHR", sexcombinations2$`Chromosome/scaffold name`), replacement, sexcombinations2$`Chromosome/scaffold name`)

rows_to_replace <- sexcombinations2$`Gene name` == "GCNT2"
sexcombinations2$`Chromosome/scaffold name`[rows_to_replace] <- "6"

rows_to_replace <- sexcombinations2$`Gene name` == "XYLT1"
sexcombinations2$`Chromosome/scaffold name`[rows_to_replace] <- "16"

sexcombinations2 <- na.omit(sexcombinations2)

sexcombinations2$ChromosomeClass <- ifelse(sexcombinations2$`Chromosome/scaffold name` %in% c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21"), "Autosomal", 
                            ifelse(sexcombinations2$`Chromosome/scaffold name` == "X", "X", "Y"))

chromosome_counts2 <- table(sexcombinations2$ChromosomeClass)

chromosome_counts_df2 <- as.data.frame(chromosome_counts2)
names(chromosome_counts_df2) <- c("ChromosomeClass", "Count")

custom_colors <- c("Autosomal" = "grey", "X" = "green", "Y" = "purple")

ggplot(chromosome_counts_df2, aes(x = "", y = Count, fill = ChromosomeClass)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start=0) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 8) +
  theme_void() +
  scale_fill_manual(values = custom_colors) + 
  labs(fill = str_wrap("Genes with sex-specific differences at baseline shared between this study and other studies",width=25))

```

#what is enriched within men and women at baseline (3.9D)
```{r}
malebaseline <- dplyr::filter(rna_compbaseline, rna_compbaseline$logFC > 0)
malebaselinegene <- rownames(malebaseline)
femalebaseline <- dplyr::filter(rna_compbaseline, rna_compbaseline$logFC < 0)
femalebaselinegene <- rownames(femalebaseline)
#extract background of all genes
allDEGs <- rownames(rna_compbaselineALL)

GO_resultsallbaseline <- enrichGO(rownames(rna_compbaseline), OrgDb = "org.Hs.eg.db", universe=allDEGs, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.2, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsmalebaseline <- enrichGO(malebaselinegene, OrgDb = "org.Hs.eg.db", universe=allDEGs, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.2, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsfemalebaseline <- enrichGO(femalebaselinegene, OrgDb = "org.Hs.eg.db", universe=allDEGs, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.2, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

df_resultmalebaseline <- as.data.frame(GO_resultsmalebaseline@result) %>% arrange(p.adjust) %>%
  head(10) %>%
  mutate(Sex = "Men")

df_resultfemalebaseline <- as.data.frame(GO_resultsfemalebaseline@result) %>% arrange(p.adjust) %>%
  head(10) %>%
  mutate(Sex = "Women")

baselineGO <- rbind(df_resultfemalebaseline, df_resultmalebaseline)

baselineGO <- baselineGO %>% mutate(Description = tools::toTitleCase(Description))

baselineGO$GeneRatio <- sapply(strsplit(baselineGO$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

baselineGO$Description <- fct_rev(baselineGO$Description)

sexstrip_background <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      Men = "#02e1f5",
      Women = "#f02bd9",
      Shared = "#e8bc74")))

#create the enrichment plot for sex specific genes at baseline
ggplot(baselineGO, aes(x = GeneRatio*100, y = Description, fill = p.adjust)) +
  geom_col() +
  scale_fill_gradientn(name = "Adjusted p-value",
                       colours = c("red", "blue"),
                       limits = c(0, 0.05),
                       sexstrip_background,
  breaks = seq(0.01, 0.05, by = 0.01), na.value = "black") +
  theme_minimal() +
  labs(y = NULL, x = "Percentage of upregulated genes") +
  ggtitle("Top ten GOBP terms with sex-specific upregulation at baseline") +
  theme(axis.text.y = element_text(size = 8), legend.position = "right",
        panel.border = element_rect(color = "black", fill = NA),
        strip.background = element_rect(fill = "grey80", color = "grey50"),
        strip.text = element_text(face = "bold", size = 10),
        axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        axis.ticks.length = unit(0.1, "cm"),
        legend.text = element_text(size = 6),
        plot.margin = ggplot2::margin(b = 0.5, unit = "cm")) +
  facet_nested(~ Sex, scales = "free_y",
               switch = "y",
               space = "free_y",
               strip = sexstrip_background) +
  guides(size = guide_legend(), color = guide_colorbar(title.position = "top"))
```

#DE of just men (same pipeline using limma)
```{r}
columnskeepdifwomen <- c(10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,82,83,84,85,86,87,88,89,90,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,246,247,248,249,250,251,252,253,254,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,	337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353)
countsmen <- counts %>% dplyr::select(-columnskeepdifwomen)
dim(countsmen)

sampledatamen <- read.xlsx("data/men20230911_sampledata-rnaseq.xlsx")
sampledatamen <- na.omit(sampledatamen)

groupmen <- t(sampledatamen[c(3)])
ymen <- DGEList(counts = countsmen, group = groupmen)

sexmen <- sampledatamen$Sex
sampleidmen <- sampledatamen$SAMPLE.ID

ymen$samples$sex <- as.factor(sexmen)
ymen$samples$sampleid <- as.factor(sampleidmen)

cpmmen <- cpm(ymen)
lcpmmen <- cpm(ymen, log=TRUE)

Lmen <- mean(ymen$samples$lib.size) * 1e-6
Mmen <- median(ymen$samples$lib.size) * 1e-6
c(Lmen, Mmen)
summary(lcpmmen)

table(rowSums(ymen$counts==0)==175)
keep.exprsmen <- filterByExpr(ymen)
#keep.exprs
ymen <- ymen[keep.exprsmen,,keep.lib.sizes=FALSE]
dim(ymen)

ymen <- calcNormFactors(ymen, method = "TMM")
ymen$samples$norm.factors

designmen <- model.matrix(~0 + group, ymen$samples)
colnames(designmen) <- gsub("group", "", colnames(designmen))

contr.matrixmen <- makeContrasts(
 "MIDvsPRE" = MID-PRE,
 "POSTvsPRE" = POST-PRE,
 "H3vsPRE" = threeH-PRE,
 "H6vsPRE" = sixH-PRE,
 "H9vsPRE" = nineH-PRE,
 "H12vsPRE" = twelveH-PRE,
 "H24vsPRE" = twentyfourH-PRE,
 "H48vsPRE" = fourtyeightH-PRE,
    levels = colnames(designmen))
contr.matrixmen

vmen <- voom(ymen, designmen, plot=TRUE)

cormen <- duplicateCorrelation(countsmen, designmen, block=sampleidmen)

fit2men <- lmFit(vmen, design=designmen, 
  block=sampleidmen, correlation=cormen$consensus.correlation)
fit2men <- contrasts.fit(fit2men, contrasts=contr.matrixmen)
efit2men <- eBayes(fit2men)

summary(decideTests(efit2men))

rna_compMEN<- topTable(efit2men, n=Inf, adjust.method = "BH")
rna_compMENALL<- topTable(efit2men, n=Inf, adjust.method = "BH")
rna_compMEN <- subset(rna_compMEN, rna_compMEN$adj.P.Val<0.05)
openxlsx::write.xlsx(rna_compMENALL,"rna_compMENALL.xlsx", rowNames = TRUE)
```

#DE of just women (same pipeline using limma)
```{r}
columnskeepdifwomen <- c(10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,82,83,84,85,86,87,88,89,90,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,246,247,248,249,250,251,252,253,254,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,	337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353)
countsdifwomen <- counts %>% dplyr::select(columnskeepdifwomen)
dim(countsdifwomen)

#sampledata read in 
sampledatadifwomen <- read.xlsx("data/women20230911_sampledata-rnaseq.xlsx")
sampledatadifwomen <- na.omit(sampledatadifwomen)

groupdifwomen <- t(sampledatadifwomen[c(3)])
ydifwomen <- DGEList(counts = countsdifwomen, group = groupdifwomen)

sexdifwomen <- sampledatadifwomen$Sex
sampleiddifwomen <- sampledatadifwomen$SAMPLE.ID

ydifwomen$samples$sex <- as.factor(sexdifwomen)
ydifwomen$samples$sampleid <- as.factor(sampleiddifwomen)

cpmdifwomen <- cpm(ydifwomen)
lcpmdifwomen <- cpm(ydifwomen, log=TRUE)

Ldifwomen <- mean(ydifwomen$samples$lib.size) * 1e-6
Mdifwomen <- median(ydifwomen$samples$lib.size) * 1e-6
c(Ldifwomen, Mdifwomen)
summary(lcpmdifwomen)

table(rowSums(ydifwomen$counts==0)==178)
keep.exprsdifwomen <- filterByExpr(ydifwomen)

ydifwomen <- ydifwomen[keep.exprsdifwomen,,keep.lib.sizes=FALSE]
dim(ydifwomen)

ydifwomen <- calcNormFactors(ydifwomen, method = "TMM")
ydifwomen$samples$norm.factors

designdifwomen <- model.matrix(~0 + group, ydifwomen$samples)
colnames(designdifwomen) <- gsub("group", "", colnames(designdifwomen))

contr.matrixdifwomen <- makeContrasts(
 "MIDvsPRE" = MID-PRE,
 "POSTvsPRE" = POST-PRE,
 "H3vsPRE" = threeH-PRE,
 "H6vsPRE" = sixH-PRE,
 "H9vsPRE" = nineH-PRE,
 "H12vsPRE" = twelveH-PRE,
 "H24vsPRE" = twentyfourH-PRE,
 "H48vsPRE" = fourtyeightH-PRE,
    levels = colnames(designdifwomen))
contr.matrixdifwomen

vdifwomen <- voom(ydifwomen, designdifwomen, plot=TRUE)

cordifwomen <- duplicateCorrelation(countsdifwomen, designdifwomen, block=sampleiddifwomen)

fit2difwomen <- lmFit(vdifwomen, design=designdifwomen, 
  block=sampleiddifwomen, correlation=cordifwomen$consensus.correlation)
fit2difwomen <- contrasts.fit(fit2difwomen, contrasts=contr.matrixdifwomen)
efit2difwomen <- eBayes(fit2difwomen)

rna_compdifwomen <- topTable(efit2difwomen, n=Inf, adjust.method = "BH")
rna_compdifwomenALL <- topTable(efit2difwomen, n=Inf, adjust.method = "BH")
rna_compdifwomen <- subset(rna_compdifwomen, rna_compdifwomen$adj.P.Val<0.05)
openxlsx::write.xlsx(rna_compdifwomenALL,"rna_compdifwomenALL.xlsx", rowNames = TRUE)
```

#extract out individual comparisions to create the plots of total exercise-induced DEGs (3.10A)
```{r}
#extract individual comparisons relative to PRE

top_treat <- function(coef, efit2) {
    result <- topTreat(efit2, coef = coef, n = Inf, adjust.method = "BH")
    result <- subset(result, result$adj.P.Val < 0.05)
    return(result)
  }
  
MIDvsPREmen <- top_treat(1, efit2men)
POSTvsPREmen <- top_treat(2, efit2men)
H3vsPREmen <- top_treat(3, efit2men)
H6vsPREmen <- top_treat(4, efit2men)
H9vsPREmen <- top_treat(5, efit2men)
H12vsPREmen <- top_treat(6, efit2men)
H24vsPREmen <- top_treat(7, efit2men)
H48vsPREmen <- top_treat(8, efit2men)
  
MIDvsPREdifwomen <- top_treat(1, efit2difwomen)
POSTvsPREdifwomen <- top_treat(2, efit2difwomen)
H3vsPREdifwomen <- top_treat(3, efit2difwomen)
H6vsPREdifwomen <- top_treat(4, efit2difwomen)
H9vsPREdifwomen <- top_treat(5, efit2difwomen)
H12vsPREdifwomen <- top_treat(6, efit2difwomen)
H24vsPREdifwomen <- top_treat(7, efit2difwomen)
H48vsPREdifwomen <- top_treat(8, efit2difwomen)

process_DE_genes <- function(datamen, datawomen, time_point, previous_genes) {
 
  DEmen_down <- datamen %>% dplyr::filter(datamen$logFC < 0)
  DEmen_down <- rownames(DEmen_down)
  
  DEdifwomen_down <- datawomen %>% dplyr::filter(datawomen$logFC < 0)
  DEdifwomen_down <- rownames(DEdifwomen_down)
  
  DEshared_down <- intersect(DEmen_down, DEdifwomen_down)
  DEonlydifwomen_down <- setdiff(DEdifwomen_down, DEmen_down)
  DEonlymen_down <- setdiff(DEmen_down, DEdifwomen_down)
  
  DEshared_down_previous <- intersect(DEshared_down, previous_genes)
  DEshared_down_previous <- length(DEshared_down_previous)
  DEshared_down_previous <- DEshared_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Shared Down", Time = time_point, Previous = "Previously identified")
  colnames(DEshared_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEshared_down_previous$Genes <- -DEshared_down_previous$Genes
  
  DEshared_down_new <- setdiff(DEshared_down, previous_genes)
  DEshared_down_new <- length(DEshared_down_new)
  DEshared_down_new <- DEshared_down_new %>% as.data.frame() %>%
    mutate(Direction = "Shared Down", Time = time_point, Previous = "Never")
  colnames(DEshared_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEshared_down_new$Genes <- -DEshared_down_new$Genes
  
  DEonlydifwomen_down_previous <- intersect(DEonlydifwomen_down, previous_genes)
  DEonlydifwomen_down_previous <- length(DEonlydifwomen_down_previous)
  DEonlydifwomen_down_previous <- DEonlydifwomen_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Women Down", Time = time_point, Previous = "Previously identified")
  colnames(DEonlydifwomen_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEonlydifwomen_down_previous$Genes <- -DEonlydifwomen_down_previous$Genes
  
  DEonlydifwomen_down_new <- setdiff(DEonlydifwomen_down, previous_genes)
  DEonlydifwomen_down_new <- length(DEonlydifwomen_down_new)
  DEonlydifwomen_down_new <- DEonlydifwomen_down_new %>% as.data.frame() %>%
    mutate(Direction = "Women Down", Time = time_point, Previous = "Never")
  colnames(DEonlydifwomen_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEonlydifwomen_down_new$Genes <- -DEonlydifwomen_down_new$Genes
  
  DEonlymen_down_previous <- intersect(DEonlymen_down, previous_genes)
  DEonlymen_down_previous <- length(DEonlymen_down_previous)
  DEonlymen_down_previous <- DEonlymen_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Men Down", Time = time_point, Previous = "Previously identified")
  colnames(DEonlymen_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEonlymen_down_previous$Genes <- -DEonlymen_down_previous$Genes
  
  DEonlymen_down_new <- setdiff(DEonlymen_down, previous_genes)
  DEonlymen_down_new <- length(DEonlymen_down_new)
  DEonlymen_down_new <- DEonlymen_down_new %>% as.data.frame() %>%
    mutate(Direction = "Men Down", Time = time_point, Previous = "Never")
  colnames(DEonlymen_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEonlymen_down_new$Genes <- -DEonlymen_down_new$Genes
  
  DEmen_up <- datamen %>% dplyr::filter(logFC > 0)
  DEmen_up <- rownames(DEmen_up)
  
  DEdifwomen_up <- datawomen %>% dplyr::filter(logFC > 0)
  DEdifwomen_up <- rownames(DEdifwomen_up)
  
  DEshared_up <- intersect(DEmen_up, DEdifwomen_up)
  DEonlydifwomen_up <- setdiff(DEdifwomen_up, DEmen_up)
  DEonlymen_up <- setdiff(DEmen_up, DEdifwomen_up)
  
  DEshared_up_previous <- intersect(DEshared_up, previous_genes)
  DEshared_up_previous <- length(DEshared_up_previous)
  DEshared_up_previous <- DEshared_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Shared Up", Time = time_point, Previous = "Previously identified")
  colnames(DEshared_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEshared_up_new <- setdiff(DEshared_up, previous_genes)
  DEshared_up_new <- length(DEshared_up_new)
  DEshared_up_new <- DEshared_up_new %>% as.data.frame() %>%
    mutate(Direction = "Shared Up", Time = time_point, Previous = "Never")
  colnames(DEshared_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlydifwomen_up_previous <- intersect(DEonlydifwomen_up, previous_genes)
  DEonlydifwomen_up_previous <- length(DEonlydifwomen_up_previous)
  DEonlydifwomen_up_previous <- DEonlydifwomen_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Women Up", Time = time_point, Previous = "Previously identified")
  colnames(DEonlydifwomen_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlydifwomen_up_new <- setdiff(DEonlydifwomen_up, previous_genes)
  DEonlydifwomen_up_new <- length(DEonlydifwomen_up_new)
  DEonlydifwomen_up_new <- DEonlydifwomen_up_new %>% as.data.frame() %>%
    mutate(Direction = "Women Up", Time = time_point, Previous = "Never")
  colnames(DEonlydifwomen_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlymen_up_previous <- intersect(DEonlymen_up, previous_genes)
  DEonlymen_up_previous <- length(DEonlymen_up_previous)
  DEonlymen_up_previous <- DEonlymen_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Men Up", Time = time_point, Previous = "Previously identified")
  colnames(DEonlymen_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlymen_up_new <- setdiff(DEonlymen_up, previous_genes)
  DEonlymen_up_new <- length(DEonlymen_up_new)
  DEonlymen_up_new <- DEonlymen_up_new %>% as.data.frame() %>%
    mutate(Direction = "Men Up", Time = time_point, Previous = "Never")
  colnames(DEonlymen_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DE_data <- rbind(DEshared_up_previous, DEonlymen_up_previous, DEonlydifwomen_up_previous,
                   DEshared_up_new, DEonlymen_up_new, DEonlydifwomen_up_new, DEshared_down_previous,
                   DEonlymen_down_previous, DEonlydifwomen_down_previous, DEshared_down_new,
                   DEonlymen_down_new, DEonlydifwomen_down_new)
  return(DE_data)
}

SexDE1 <- process_DE_genes(MIDvsPREmen, MIDvsPREdifwomen, "MID", previous_genes)
SexDE2 <- process_DE_genes(POSTvsPREmen, POSTvsPREdifwomen, "POST", previous_genes)
SexDE3 <- process_DE_genes(H3vsPREmen, H3vsPREdifwomen, "3H", previous_genes)
SexDE4 <- process_DE_genes(H6vsPREmen, H6vsPREdifwomen, "6H", previous_genes)
SexDE5 <- process_DE_genes(H9vsPREmen, H9vsPREdifwomen, "9H", previous_genes)
SexDE6 <- process_DE_genes(H12vsPREmen, H12vsPREdifwomen, "12H", previous_genes)
SexDE7 <- process_DE_genes(H24vsPREmen, H24vsPREdifwomen, "24H", previous_genes)
SexDE8 <- process_DE_genes(H48vsPREmen, H48vsPREdifwomen, "48H", previous_genes)

SexDE <- rbind (SexDE1, SexDE2, SexDE3, SexDE4, SexDE5, SexDE6, SexDE7, SexDE8)

SexDE$Time <- as.character(SexDE$Time)
SexDE$Time <- factor(SexDE$Time, levels = unique(SexDE$Time))

# plot of total DEGs
ggplot(data = SexDE, aes(x=Time, y=Genes, fill = Direction, pattern = Previous)) + 
  geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
  labs(x = "Time", y = "No. DE genes relative to PRE", fill = "Sets of genes", pattern =str_wrap("Previously identified as exercise-induced DEG", width = 25)) +
  scale_y_continuous(name = expression("        No. DEGs relative to PRE\n \n Downregulated              Upregulated"),
                       labels = function(x) abs(x), limits=c(-6000,6000)) +
  scale_fill_manual(values = c(
      "Men Down" = "#34d8eb",
      "Men Up" = "#94e2eb",
      "Women Down" = "#e887e3",
      "Women Up" = "#e3b3e0",
      "Shared Down" = "#e8bc74",
      "Shared Up" = "#f0d5aa"), 
                        labels = c("Men Up"= "Upregulated in men", "Men Down" = "Downregulated in men",
               "Women Down" = "Downregulated in women", "Women Up" = "Upregulated in women",
               "Shared Up" = "Upregulated in both sexes", "Shared Down" = "Downregulated in both sexes")) +
  guides(fill = guide_legend(reverse = FALSE)) +
  theme_bw() +
  scale_pattern_manual(values = c("Never" = "none", "Previously identified" = "stripe"),
                           labels = c("Previously identified" = "Yes", "Never" = "No")) +
  guides(color = guide_legend(override.aes = list(pattern = "none", fill = NA)), 
  fill = guide_legend(reverse=TRUE, override.aes = list(pattern = "none", color=NA)),
         pattern = guide_legend(reverse=TRUE,override.aes = list(fill = "white"))) +
  theme(axis.title.y = element_text(size=12, margin = ggplot2::margin(r = 5, l = 20)),
            axis.title.y.right = element_text(margin = ggplot2::margin(l = 5, r =5))) +
  ggtitle("Exercise-induced DEGs in men and women")
```

#create ridge plots for sex-specific exercise-induced DEG expression (3.10B)
```{r}
add_sex_time <- function(data, sex, time) {
  mutate(data, sex = sex, time = time)
}

# Add sex and time columns for each time point
MIDvsPREmen <- add_sex_time(MIDvsPREmen, "Men", "MID")
MIDvsPREdifwomen <- add_sex_time(MIDvsPREdifwomen, "Women", "MID")
POSTvsPREmen <- add_sex_time(POSTvsPREmen, "Men", "POST")
POSTvsPREdifwomen <- add_sex_time(POSTvsPREdifwomen, "Women", "POST")
H3vsPREmen <- add_sex_time(H3vsPREmen, "Men", "3H")
H3vsPREdifwomen <- add_sex_time(H3vsPREdifwomen, "Women", "3H")
H6vsPREmen <- add_sex_time(H6vsPREmen, "Men", "6H")
H6vsPREdifwomen <- add_sex_time(H6vsPREdifwomen, "Women", "6H")
H9vsPREmen <- add_sex_time(H9vsPREmen, "Men", "9H")
H9vsPREdifwomen <- add_sex_time(H9vsPREdifwomen, "Women", "9H")
H12vsPREmen <- add_sex_time(H12vsPREmen, "Men", "12H")
H12vsPREdifwomen <- add_sex_time(H12vsPREdifwomen, "Women", "12H")
H24vsPREmen <- add_sex_time(H24vsPREmen, "Men", "24H")
H24vsPREdifwomen <- add_sex_time(H24vsPREdifwomen, "Women", "24H")
H48vsPREmen <- add_sex_time(H48vsPREmen, "Men", "48H")
H48vsPREdifwomen <- add_sex_time(H48vsPREdifwomen, "Women", "48H")

# Combine all data
combined_data <- bind_rows(MIDvsPREmen, MIDvsPREdifwomen, POSTvsPREmen, POSTvsPREdifwomen, 
                           H3vsPREmen, H3vsPREdifwomen, H6vsPREmen, H6vsPREdifwomen,
                           H9vsPREmen, H9vsPREdifwomen, H12vsPREmen, H12vsPREdifwomen,
                           H24vsPREmen, H24vsPREdifwomen, H48vsPREmen, H48vsPREdifwomen)

time_order <- c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")

# Convert time to factor with the defined order
combined_data$time <- factor(combined_data$time, levels = time_order, ordered = TRUE)

ggplot(combined_data, aes(x = logFC, y = time, fill = sex)) +
  geom_density_ridges(alpha = 0.7,scale = 1, rel_min_height = 0.01) +
  scale_fill_manual(values = c("Men" = "#02e1f5", "Women" = "#f02bd9")) +
  labs(x = bquote(Log[2] * FC ~ "relative to PRE of exercise-induced DEGs"), y = "Time", fill = NULL) +
  theme_bw() +
  theme(axis.title.x = element_text(size = 12),
        axis.text.x = element_text(size = 10),
        axis.title.y = element_blank(),
        legend.position = "right",
        legend.title = element_blank(),
        legend.text = element_text(size = 12),
        axis.ticks.y = element_blank()) +
  ggtitle("Ridge plot of sex-specific exercise-induced gene expression") +
  geom_rug(aes(y = time), sides = "l") +
  geom_rug(aes(y = time), sides = "r") +
  scale_x_continuous(breaks = seq(-5, 5, by = 1), limits = c(-3, 5)) +
  scale_y_discrete(expand = expansion(mult = c(0.1, 0.1)), limits = rev(levels(combined_data$time)))

graph2doc(file="plot1.docx", width=8,height=4.5)

```

#enrichment to see what is the most different between genes upregulated/downregulated between men and women and what is most shared (3.10C)
```{r}
genesMvW <- intersect(rownames(rna_compMENALL), rownames(rna_compdifwomenALL))
genesMvW <- unique(genesMvW)
MengenesMvW <- rownames(rna_compMENALL)
WomengenesMvW <- rownames(rna_compdifwomenALL)

MvsW_individual_timepointsUP <- function(data1, data2, interaction) {
  
  genes_data1 <- rownames(data1[data1$logFC > 0, ])
  genes_data2 <- rownames(data2[data2$logFC > 0, ])

  unique_genes <- interaction(genes_data1, genes_data2)
  
  filtered_rna_comp <- data1[rownames(data1) %in% unique_genes, ]

  DE_data <- rownames(filtered_rna_comp)
   return(DE_data)
}

MvsW_individual_timepointsDOWN <- function(data1, data2, interaction) {
 
  genes_data1 <- rownames(data1[data1$logFC < 0, ])
  genes_data2 <- rownames(data2[data2$logFC < 0, ])

  unique_genes <- interaction(genes_data1, genes_data2)
  
  filtered_rna_comp <- data1[rownames(data1) %in% unique_genes, ]
  
  DE_data <- rownames(filtered_rna_comp)
   return(DE_data)
}

MIDDOWNdifmen <- MvsW_individual_timepointsDOWN(MIDvsPREmen, MIDvsPREdifwomen, setdiff)
POSTDOWNdifmen <- MvsW_individual_timepointsDOWN(POSTvsPREmen, POSTvsPREdifwomen, setdiff)
H3DOWNdifmen <- MvsW_individual_timepointsDOWN(H3vsPREmen, H3vsPREdifwomen, setdiff)
H6DOWNdifmen <- MvsW_individual_timepointsDOWN(H6vsPREmen, H6vsPREdifwomen, setdiff)
H9DOWNdifmen <- MvsW_individual_timepointsDOWN(H9vsPREmen, H9vsPREdifwomen, setdiff)
H12DOWNdifmen <- MvsW_individual_timepointsDOWN(H12vsPREmen, H12vsPREdifwomen, setdiff)
H24DOWNdifmen <- MvsW_individual_timepointsDOWN(H24vsPREmen, H24vsPREdifwomen, setdiff)
H48DOWNdifmen <- MvsW_individual_timepointsDOWN(H48vsPREmen, H48vsPREdifwomen, setdiff)

DifMENDOWNMvsW <- c(MIDDOWNdifmen, POSTDOWNdifmen, H3DOWNdifmen, H6DOWNdifmen, H9DOWNdifmen, H12DOWNdifmen,  H24DOWNdifmen, H48DOWNdifmen)
DifMENDOWNMvsW <- unique(DifMENDOWNMvsW)

MIDUPdifmen <- MvsW_individual_timepointsUP(MIDvsPREmen, MIDvsPREdifwomen, setdiff)
POSTUPdifmen <- MvsW_individual_timepointsUP(POSTvsPREmen, POSTvsPREdifwomen, setdiff)
H3UPdifmen <- MvsW_individual_timepointsUP(H3vsPREmen, H3vsPREdifwomen, setdiff)
H6UPdifmen <- MvsW_individual_timepointsUP(H6vsPREmen, H6vsPREdifwomen, setdiff)
H9UPdifmen <- MvsW_individual_timepointsUP(H9vsPREmen, H9vsPREdifwomen, setdiff)
H12UPdifmen <- MvsW_individual_timepointsUP(H12vsPREmen, H12vsPREdifwomen, setdiff)
H24UPdifmen <- MvsW_individual_timepointsUP(H24vsPREmen, H24vsPREdifwomen, setdiff)
H48UPdifmen <- MvsW_individual_timepointsUP(H48vsPREmen, H48vsPREdifwomen, setdiff)

DifMENUPMvsW <- c(MIDUPdifmen, POSTUPdifmen, H3UPdifmen, H6UPdifmen, H9UPdifmen, H12UPdifmen,  H24UPdifmen, H48UPdifmen)
DifMENUPMvsW <- unique(DifMENUPMvsW)

MIDDOWNdifdifwomen <- MvsW_individual_timepointsDOWN(MIDvsPREdifwomen, MIDvsPREmen, setdiff)
POSTDOWNdifdifwomen <- MvsW_individual_timepointsDOWN(POSTvsPREdifwomen, POSTvsPREmen, setdiff)
H3DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H3vsPREdifwomen, H3vsPREmen, setdiff)
H6DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H6vsPREdifwomen, H6vsPREmen, setdiff)
H9DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H9vsPREdifwomen, H9vsPREmen, setdiff)
H12DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H12vsPREdifwomen, H12vsPREmen, setdiff)
H24DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H24vsPREdifwomen, H24vsPREmen, setdiff)
H48DOWNdifdifwomen <- MvsW_individual_timepointsDOWN(H48vsPREdifwomen, H48vsPREmen, setdiff)

DifdifwomenDOWNMvsW <- c(MIDDOWNdifdifwomen, POSTDOWNdifdifwomen, H3DOWNdifdifwomen, H6DOWNdifdifwomen, H9DOWNdifdifwomen, H12DOWNdifdifwomen,  H24DOWNdifdifwomen, H48DOWNdifdifwomen)
DifdifwomenDOWNMvsW <- unique(DifdifwomenDOWNMvsW)

plug <- intersect(DifdifwomenDOWNMvsW, rownames(rna_compMEN))

MIDUPdifdifwomen <- MvsW_individual_timepointsUP(MIDvsPREdifwomen, MIDvsPREmen, setdiff)
POSTUPdifdifwomen <- MvsW_individual_timepointsUP(POSTvsPREdifwomen, POSTvsPREmen, setdiff)
H3UPdifdifwomen <- MvsW_individual_timepointsUP(H3vsPREdifwomen, H3vsPREmen, setdiff)
H6UPdifdifwomen <- MvsW_individual_timepointsUP(H6vsPREdifwomen, H6vsPREmen, setdiff)
H9UPdifdifwomen <- MvsW_individual_timepointsUP(H9vsPREdifwomen, H9vsPREmen, setdiff)
H12UPdifdifwomen <- MvsW_individual_timepointsUP(H12vsPREdifwomen, H12vsPREmen, setdiff)
H24UPdifdifwomen <- MvsW_individual_timepointsUP(H24vsPREdifwomen, H24vsPREmen, setdiff)
H48UPdifdifwomen <- MvsW_individual_timepointsUP(H48vsPREdifwomen, H48vsPREmen, setdiff)

DifdifwomenUPMvsW <- c(MIDUPdifdifwomen, POSTUPdifdifwomen, H3UPdifdifwomen, H6UPdifdifwomen, H9UPdifdifwomen, H12UPdifdifwomen,  H24UPdifdifwomen, H48UPdifdifwomen)
DifdifwomenUPMvsW <- unique(DifdifwomenUPMvsW)

MIDDOWNSHARED <- MvsW_individual_timepointsDOWN(MIDvsPREdifwomen, MIDvsPREmen, intersect)
POSTDOWNSHARED <- MvsW_individual_timepointsDOWN(POSTvsPREdifwomen, POSTvsPREmen, intersect)
H3DOWNSHARED <- MvsW_individual_timepointsDOWN(H3vsPREdifwomen, H3vsPREmen, intersect)
H6DOWNSHARED <- MvsW_individual_timepointsDOWN(H6vsPREdifwomen, H6vsPREmen, intersect)
H9DOWNSHARED <- MvsW_individual_timepointsDOWN(H9vsPREdifwomen, H9vsPREmen, intersect)
H12DOWNSHARED <- MvsW_individual_timepointsDOWN(H12vsPREdifwomen, H12vsPREmen, intersect)
H24DOWNSHARED <- MvsW_individual_timepointsDOWN(H24vsPREdifwomen, H24vsPREmen, intersect)
H48DOWNSHARED <- MvsW_individual_timepointsDOWN(H48vsPREdifwomen, H48vsPREmen, intersect)

SHAREDDOWNMvsW <- c(MIDDOWNSHARED, POSTDOWNSHARED, H3DOWNSHARED, H6DOWNSHARED, H9DOWNSHARED, H12DOWNSHARED,  H24DOWNSHARED, H48DOWNSHARED)
SHAREDDOWNMvsW <- unique(SHAREDDOWNMvsW)

MIDUPSHARED <- MvsW_individual_timepointsUP(MIDvsPREdifwomen, MIDvsPREmen, intersect)
POSTUPSHARED <- MvsW_individual_timepointsUP(POSTvsPREdifwomen, POSTvsPREmen, intersect)
H3UPSHARED <- MvsW_individual_timepointsUP(H3vsPREdifwomen, H3vsPREmen, intersect)
H6UPSHARED <- MvsW_individual_timepointsUP(H6vsPREdifwomen, H6vsPREmen, intersect)
H9UPSHARED <- MvsW_individual_timepointsUP(H9vsPREdifwomen, H9vsPREmen, intersect)
H12UPSHARED <- MvsW_individual_timepointsUP(H12vsPREdifwomen, H12vsPREmen, intersect)
H24UPSHARED <- MvsW_individual_timepointsUP(H24vsPREdifwomen, H24vsPREmen, intersect)
H48UPSHARED <- MvsW_individual_timepointsUP(H48vsPREdifwomen, H48vsPREmen, intersect)

SHAREDUPMvsW <- c(MIDUPSHARED, POSTUPSHARED, H3UPSHARED, H6UPSHARED, H9UPSHARED, H12UPSHARED,  H24UPSHARED, H48UPSHARED)
SHAREDUPMvsW <- unique(SHAREDUPMvsW)

GO_resultsMUP <- enrichGO(DifMENUPMvsW, OrgDb = "org.Hs.eg.db", universe=MengenesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsWUP <- enrichGO(DifdifwomenUPMvsW, OrgDb = "org.Hs.eg.db", universe=WomengenesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsMDOWN <- enrichGO(DifMENDOWNMvsW, OrgDb = "org.Hs.eg.db", universe=MengenesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsWDOWN <- enrichGO(DifdifwomenDOWNMvsW, OrgDb = "org.Hs.eg.db", universe=WomengenesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsSHAREDUP <- enrichGO(SHAREDUPMvsW, OrgDb = "org.Hs.eg.db", universe=genesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

GO_resultsSHAREDDOWN <- enrichGO(SHAREDDOWNMvsW, OrgDb = "org.Hs.eg.db", universe=genesMvW, keyType = "SYMBOL", ont = "BP",pvalueCutoff = 0.05, qvalueCutoff = 0.05, pAdjustMethod = "BH", readable = FALSE, pool = FALSE)

df_MENUP <- as.data.frame(GO_resultsMUP@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Men") %>% mutate(Direction = "UP")

df_difwomenUP <- as.data.frame(GO_resultsWUP@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Women") %>% mutate(Direction = "UP")

df_MENDOWN <- as.data.frame(GO_resultsMDOWN@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Men") %>% mutate(Direction = "DOWN")

df_difwomenDOWN <- as.data.frame(GO_resultsWDOWN@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Women") %>% mutate(Direction = "DOWN")

df_SHAREDUP <- as.data.frame(GO_resultsSHAREDUP@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Shared") %>% mutate(Direction = "UP")

df_SHAREDDOWN <- as.data.frame(GO_resultsSHAREDDOWN@result) %>% arrange(p.adjust) %>%
  head(5) %>% mutate(Sex = "Shared") %>% mutate(Direction = "DOWN")

MvsWchanges <- rbind(df_MENUP, df_difwomenUP, df_MENDOWN, df_difwomenDOWN, df_SHAREDUP, df_SHAREDDOWN)

MvsWchanges <- MvsWchanges %>% mutate(Description = tools::toTitleCase(Description))
MvsWchanges$Description <- fct_rev(MvsWchanges$Description)

MvsWchanges$GeneRatio <- sapply(strsplit(MvsWchanges$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

Direction_orderMvsW <- c("UP", "DOWN")
MvsWchanges$Direction <- factor(MvsWchanges$Direction, levels = Direction_orderMvsW)

Sex_orderMvsW <- c("Men", "Women", "Shared")
MvsWchanges$Sex <- factor(MvsWchanges$Sex, levels = Sex_orderMvsW)

MvsWchanges <- MvsWchanges %>% dplyr::filter(MvsWchanges$p.adjust<0.05)

sexstrip_background <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      Men = "#02e1f5",
      Women = "#f02bd9",
      Shared = "#e8bc74"
    )
  )
)

ggplot(MvsWchanges, aes(x = Direction, y = Description, size = GeneRatio*100, color = p.adjust)) +
  geom_point() +
  scale_size_continuous(name = "% of genes", range = c(2, 4)) +
  scale_color_gradientn(name = "Adjusted p-value", colours = c("red", "blue"), limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01), na.value = "grey50") +
  theme_minimal() +
  labs(y = NULL) +
  ggtitle("Top five GOBP terms of exercise-induced DEGs different and shared between each sex") +
  theme(axis.text.y = element_text(size = 8), legend.position = "right",
        panel.border = element_rect(color = "black", fill = NA),
        strip.background = element_rect(fill = "grey80", color = "grey50"),
        strip.text = element_text(face = "bold", size = 10),
         axis.title.x = element_blank(),
                axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
    axis.ticks.length = unit(0.1, "cm"),
    legend.text = element_text(size = 8)
  ) +
  scale_x_discrete(labels=c("UP" = "Upregulated", "DOWN"="Downregulated")) +
  facet_nested(. ~ Sex, strip = sexstrip_background) +
  guides(size = guide_legend(), color = guide_colorbar(title.position = "top"))

```

#having a deeper look at mitochondrial gene expression differences between men and women, shown as euler plot (3.11A)
```{r}
#creating a euler plot of what is shared/different between men and women
#get mitoDEGs for men and women
mitomen <- merge(rna_compMEN, mitocarta, by = "Symbol", "row.names", keep.rownames = TRUE)
mitomen <- as.data.frame(mitomen[,-1], row.names=mitomen[,1])
mitomen <- dplyr::select(mitomen, 1:8)

mitodifwomen <- merge(rna_compdifwomen, mitocarta, by = "Symbol", "row.names", keep.rownames = TRUE)
mitodifwomen <- as.data.frame(mitodifwomen[,-1], row.names=mitodifwomen[,1])
mitodifwomen <- dplyr::select(mitodifwomen, 1:8)


setdiff(rownames(mitomen), mitoDE$Row.names)

color_palettesex <- c(
  "Mitochondria" = "#d65c5c", 
  "Men" = "#94e2eb", 
  "Women" = "#e3b3e0", 
  "OnlyWomen" = "#e887e3", 
  "OnlyMen" = "#34d8eb", 
  "Mitochondria&Men&Women" = "orange"
)

Eulersexmito <- eulerr::euler(
  c(
    "Mitochondria" = 1136, 
    "Men" = 0, 
    "Women" = 0, 
    "OnlyWomen" = 0, 
    "OnlyMen" = 0, 
    "Mitochondria&Men" = 58, 
    "Mitochondria&Women" = 57, 
    "Mitochondria&Men&Women" = 858 
  ),
  shape = "circle"
)

plot(
  Eulersexmito, 
  quantities = TRUE,
  fill = color_palettesex,
  lty = 0,
  labels = NULL,
  main = NULL,
  legend = list(labels = c(
    "All mitochondrial genes", 
    "Differentially expressed in only men", 
    "Differentially expressed in only women"
  ))
)

```

#Graph of sex-specific DE mito genes across time points (3.11B)
```{r}
Mitogenes <- Gomit$Symbol
  
process_subset_DE_genes <- function(datamen, datawomen, time_point, previous_genes, Mitogenes) {
 
  DEmen_down <- datamen %>% dplyr::filter(datamen$logFC < 0)
  DEmen_down <- rownames(DEmen_down)[rownames(DEmen_down) %in% Mitogenes]
  
  DEdifwomen_down <- datawomen %>% dplyr::filter(datawomen$logFC < 0)
  DEdifwomen_down <- rownames(DEdifwomen_down)[rownames(DEdifwomen_down) %in% Mitogenes]
  
  DEshared_down <- intersect(DEmen_down, DEdifwomen_down)
  DEonlydifwomen_down <- setdiff(DEdifwomen_down, DEmen_down)
  DEonlymen_down <- setdiff(DEmen_down, DEdifwomen_down)
  
  DEshared_down_previous <- intersect(DEshared_down, previous_genes)
  DEshared_down_previous <- length(DEshared_down_previous)
  DEshared_down_previous <- DEshared_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Shared Down", Time = time_point, Previous = "Previously identified")
  colnames(DEshared_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEshared_down_previous$Genes <- -DEshared_down_previous$Genes
  
  DEshared_down_new <- setdiff(DEshared_down, previous_genes)
  DEshared_down_new <- length(DEshared_down_new)
  DEshared_down_new <- DEshared_down_new %>% as.data.frame() %>%
    mutate(Direction = "Shared Down", Time = time_point, Previous = "Never")
  colnames(DEshared_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEshared_down_new$Genes <- -DEshared_down_new$Genes
  
  DEonlydifwomen_down_previous <- intersect(DEonlydifwomen_down, previous_genes)
  DEonlydifwomen_down_previous <- length(DEonlydifwomen_down_previous)
  DEonlydifwomen_down_previous <- DEonlydifwomen_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Women Down", Time = time_point, Previous = "Previously identified")
  colnames(DEonlydifwomen_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEonlydifwomen_down_previous$Genes <- -DEonlydifwomen_down_previous$Genes
  
  DEonlydifwomen_down_new <- setdiff(DEonlydifwomen_down, previous_genes)
  DEonlydifwomen_down_new <- length(DEonlydifwomen_down_new)
  DEonlydifwomen_down_new <- DEonlydifwomen_down_new %>% as.data.frame() %>%
    mutate(Direction = "Women Down", Time = time_point, Previous = "Never")
  colnames(DEonlydifwomen_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEonlydifwomen_down_new$Genes <- -DEonlydifwomen_down_new$Genes
  
  DEonlymen_down_previous <- intersect(DEonlymen_down, previous_genes)
  DEonlymen_down_previous <- length(DEonlymen_down_previous)
  DEonlymen_down_previous <- DEonlymen_down_previous %>% as.data.frame() %>%
    mutate(Direction = "Men Down", Time = time_point, Previous = "Previously identified")
  colnames(DEonlymen_down_previous) <- c("Genes", "Direction", "Time", "Previous")
  DEonlymen_down_previous$Genes <- -DEonlymen_down_previous$Genes
  
  DEonlymen_down_new <- setdiff(DEonlymen_down, previous_genes)
  DEonlymen_down_new <- length(DEonlymen_down_new)
  DEonlymen_down_new <- DEonlymen_down_new %>% as.data.frame() %>%
    mutate(Direction = "Men Down", Time = time_point, Previous = "Never")
  colnames(DEonlymen_down_new) <- c("Genes", "Direction", "Time", "Previous")
  DEonlymen_down_new$Genes <- -DEonlymen_down_new$Genes
  
  DEmen_up <- datamen %>% dplyr::filter(logFC > 0)
  DEmen_up <- rownames(DEmen_up)[rownames(DEmen_up) %in% Mitogenes]
  
  DEdifwomen_up <- datawomen %>% dplyr::filter(logFC > 0)
  DEdifwomen_up <- rownames(DEdifwomen_up)[rownames(DEdifwomen_up) %in% Mitogenes]
  
  DEshared_up <- intersect(DEmen_up, DEdifwomen_up)
  DEonlydifwomen_up <- setdiff(DEdifwomen_up, DEmen_up)
  DEonlymen_up <- setdiff(DEmen_up, DEdifwomen_up)
  
  DEshared_up_previous <- intersect(DEshared_up, previous_genes)
  DEshared_up_previous <- length(DEshared_up_previous)
  DEshared_up_previous <- DEshared_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Shared Up", Time = time_point, Previous = "Previously identified")
  colnames(DEshared_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEshared_up_new <- setdiff(DEshared_up, previous_genes)
  DEshared_up_new <- length(DEshared_up_new)
  DEshared_up_new <- DEshared_up_new %>% as.data.frame() %>%
    mutate(Direction = "Shared Up", Time = time_point, Previous = "Never")
  colnames(DEshared_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlydifwomen_up_previous <- intersect(DEonlydifwomen_up, previous_genes)
  DEonlydifwomen_up_previous <- length(DEonlydifwomen_up_previous)
  DEonlydifwomen_up_previous <- DEonlydifwomen_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Women Up", Time = time_point, Previous = "Previously identified")
  colnames(DEonlydifwomen_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlydifwomen_up_new <- setdiff(DEonlydifwomen_up, previous_genes)
  DEonlydifwomen_up_new <- length(DEonlydifwomen_up_new)
  DEonlydifwomen_up_new <- DEonlydifwomen_up_new %>% as.data.frame() %>%
    mutate(Direction = "Women Up", Time = time_point, Previous = "Never")
  colnames(DEonlydifwomen_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlymen_up_previous <- intersect(DEonlymen_up, previous_genes)
  DEonlymen_up_previous <- length(DEonlymen_up_previous)
  DEonlymen_up_previous <- DEonlymen_up_previous %>% as.data.frame() %>%
    mutate(Direction = "Men Up", Time = time_point, Previous = "Previously identified")
  colnames(DEonlymen_up_previous) <- c("Genes", "Direction", "Time", "Previous")
  
  DEonlymen_up_new <- setdiff(DEonlymen_up, previous_genes)
  DEonlymen_up_new <- length(DEonlymen_up_new)
  DEonlymen_up_new <- DEonlymen_up_new %>% as.data.frame() %>%
    mutate(Direction = "Men Up", Time = time_point, Previous = "Never")
  colnames(DEonlymen_up_new) <- c("Genes", "Direction", "Time", "Previous")
  
  DE_data <- rbind(DEshared_up_previous, DEonlymen_up_previous, DEonlydifwomen_up_previous, DEshared_up_new, DEonlymen_up_new, DEonlydifwomen_up_new, DEshared_down_previous, DEonlymen_down_previous, DEonlydifwomen_down_previous, DEshared_down_new, DEonlymen_down_new, DEonlydifwomen_down_new)
  return(DE_data)
}

SexmitoDE1 <- process_subset_DE_genes(MIDvsPREmen, MIDvsPREdifwomen, "MID", previous_genes, Mitogenes)
SexmitoDE2 <- process_subset_DE_genes(POSTvsPREmen, POSTvsPREdifwomen, "POST", previous_genes, Mitogenes)
SexmitoDE3 <- process_subset_DE_genes(H3vsPREmen, H3vsPREdifwomen, "3H", previous_genes, Mitogenes)
SexmitoDE4 <- process_subset_DE_genes(H6vsPREmen, H6vsPREdifwomen, "6H", previous_genes, Mitogenes)
SexmitoDE5 <- process_subset_DE_genes(H9vsPREmen, H9vsPREdifwomen, "9H", previous_genes, Mitogenes)
SexmitoDE6 <- process_subset_DE_genes(H12vsPREmen, H12vsPREdifwomen, "12H", previous_genes, Mitogenes)
SexmitoDE7 <- process_subset_DE_genes(H24vsPREmen, H24vsPREdifwomen, "24H", previous_genes, Mitogenes)
SexmitoDE8 <- process_subset_DE_genes(H48vsPREmen, H48vsPREdifwomen, "48H", previous_genes, Mitogenes)

SexmitoDE <- rbind (SexmitoDE1, SexmitoDE2, SexmitoDE3, SexmitoDE4, SexmitoDE5, SexmitoDE6, SexmitoDE7, SexmitoDE8)

SexmitoDE$Time <- as.character(SexmitoDE$Time)
SexmitoDE$Time <- factor(SexmitoDE$Time, levels = unique(SexmitoDE$Time))

# Use the correct levels in the scale_fill_manual
ggplot(data = SexmitoDE, aes(x=Time, y=Genes, fill = Direction, pattern = Previous)) + 
       geom_col_pattern(color = "black", 
                   pattern_fill = "black",
                   pattern_angle = 45,
                   pattern_density = 0.1,
                   pattern_spacing = 0.025,
                   pattern_key_scale_factor = 0.6) +
      labs(x = "Time", y = "No. DE genes relative to PRE", fill = "Sets of genes", pattern =str_wrap("Previously identified as exercise-induced DEG", width = 25)) +
      scale_y_continuous(name = expression("        No. DEGs relative to PRE\n \n Downregulated              Upregulated"),
                      labels = function(x) abs(x), limits=c(-600,400)) +
      scale_fill_manual(name = "Sets of genes",
                        values = c("Men Down" = "#34d8eb",
                                  "Men Up" = "#94e2eb",
                                  "Women Down" = "#e887e3",
                                  "Women Up" = "#e3b3e0",
                                  "Shared Down" = "#b03333",
                                  "Shared Up" = "#ed3434"), 
                        labels = c("Men Up"= "Upregulated in men", "Men Down" = "Downregulated in men",
               "Women Down" = "Downregulated in women", "Women Up" = "Upregulated in women",
               "Shared Up" = "Upregulated in both sexes", "Shared Down" = "Downregulated in both sexes")) +
      guides(fill = guide_legend(reverse = TRUE)) +
      theme_bw() +
      scale_pattern_manual(values = c("Never" = "none", "Previously identified" = "stripe"),
                           labels = c("Previously identified" = "Yes", "Never" = "No")) +
      guides(color = guide_legend(override.aes = list(pattern = "none", fill = NA)), 
             fill = guide_legend(reverse=TRUE,override.aes = list(pattern = "none", color=NA)),
             pattern = guide_legend(reverse=TRUE, override.aes = list(fill = "white"))) +
      theme(axis.title.y = element_text(margin = ggplot2::margin(r = 5, l = 20)),
            axis.title.y.right = element_text(margin = ggplot2::margin(l = 5, r =5))) +
      ggtitle("Exercise-induced mitochondrial DEGs between men and women")
```

#deeper look at those mito genes which are only differentially expressed in either sex (heatmaps not used, only dataframes used for later enrichment)
```{r}
#get mitoDEGs for men and women
mitocarta <- read.xlsx("data/MitoCarta30.xlsx")

mitomen <- merge(rna_compMEN, mitocarta, by = "Symbol", "row.names", keep.rownames = TRUE)
mitomen <- as.data.frame(mitomen[,-1], row.names=mitomen[,1])
mitomen <- dplyr::select(mitomen, 1:8)

mitodifwomen <- merge(rna_compdifwomen, mitocarta, by = "Symbol", "row.names", keep.rownames = TRUE)
mitodifwomen <- as.data.frame(mitodifwomen[,-1], row.names=mitodifwomen[,1])
mitodifwomen <- dplyr::select(mitodifwomen, 1:8)

#identify what DEG genes are unique to men and women
notDEexpressedmen <- setdiff(rownames(mitodifwomen), rownames(mitomen))
notDEexpresseddifwomen <- setdiff(rownames(mitomen), rownames(mitodifwomen))

#what DEGs are shared (853 genes)
sharedDEGs <- intersect(rownames(mitodifwomen), rownames(mitomen))

#what is not expressed at all (2 unique for men, 1 for women)
notexpressedmen <- setdiff(rownames(mitodifwomen), rownames(rna_compMENALL))
notexpresseddifwomen <- setdiff(rownames(mitomen), rownames(rna_compdifwomenALL))

#creating a heatmap of the unique differentially expressed genes
colramp <- colorRamp2(c(-2, 0, 2), c("#0A3C72", "white", "#7B0722"))
labelsHM <- expression(Log[2]*FC)

#for women

mitoDEonlywomen <- subset(rna_compdifwomen, row.names(rna_compdifwomen) %in% notDEexpressedmen)
mitoDEonlywomendend <- mitoDEonlywomen[c(1:8)] %>% dist(method = "euclidean") %>% hclust(method = "average") %>% as.dendrogram

subset_df <- mitoDEonlywomen[, 1:8]
above_0_rows <- subset_df[rowSums(subset_df > 0) > 0, ]
below_0_rows <- subset_df[rowSums(subset_df < 0) > 0, ]
max_values_above_0 <- apply(above_0_rows, 1, max)
min_values_below_0 <- apply(below_0_rows, 1, min)
average_max <- mean(max_values_above_0)
average_min <- mean(min_values_below_0)

Heatmap(as.matrix(mitoDEonlywomen[c(1:8)]), heatmap_legend_param = list(title = labelsHM, legend_width = unit(6, "cm")), col=colramp, row_labels = rownames(mitoDEonlywomen), cluster_rows = mitoDEonlywomendend, show_column_dend = FALSE, show_row_names = TRUE, show_row_dend = FALSE, cluster_columns =FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 6), row_names_side = "left", column_labels = c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H"), column_names_rot = 45)

#for men
mitoDEonlymen <- subset(rna_compMEN, row.names(rna_compMEN) %in% notDEexpresseddifwomen)
mitoDEonlymendend <- mitoDEonlymen[c(1:8)] %>% dist(method = "euclidean") %>% hclust(method = "average") %>% as.dendrogram

subset_df <- mitoDEonlymen[, 1:8]
above_0_rows <- subset_df[rowSums(subset_df > 0) > 0, ]
below_0_rows <- subset_df[rowSums(subset_df < 0) > 0, ]
max_values_above_0 <- apply(above_0_rows, 1, max)
min_values_below_0 <- apply(below_0_rows, 1, min)
average_max <- mean(max_values_above_0)
average_min <- mean(min_values_below_0)

Heatmap(as.matrix(mitoDEonlymen[c(1:8)]), heatmap_legend_param = list(title = labelsHM, legend_width = unit(6, "cm")), col=colramp, row_labels = rownames(mitoDEonlymen), cluster_rows = mitoDEonlymendend, show_column_dend = FALSE, show_row_names = TRUE, show_row_dend = FALSE, cluster_columns =FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 6), row_names_side = "left", column_labels = c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H"), column_names_rot = 45)

#creating an enrichment of these 49 genes for men and 59 for women to use for the next chunk - background is genes detected within either men and women ?

mitocartaadj <- read.xlsx("data/mitocarta30adj.xlsx")
colnames(mitocartaadj) <- c("term", "gene", "hier")
mitocartaadj <- mitocartaadj %>%
  mutate(gene = strsplit(as.character(gene), ", ")) %>%
  unnest(cols = gene)

mitocartaenrichonlymen <- enricher(notDEexpresseddifwomen,  universe = rownames(mitomen), pAdjustMethod = "BH", gson = NULL, TERM2GENE = mitocartaadj)
mitocartaenrichonlymen <- as.data.frame(mitocartaenrichonlymen@result) %>% arrange(pvalue) %>% mutate(Sex = "Only DE in men") 

mitocartaenrichonlydifwomen <- enricher(notDEexpressedmen, universe = rownames(mitodifwomen), pAdjustMethod = "BH", gson = NULL, TERM2GENE = mitocartaadj)
mitocartaenrichonlydifwomen <- as.data.frame(mitocartaenrichonlydifwomen@result) %>% arrange(pvalue) %>% mutate(Sex = "Only DE in women") 
```

#looking at percentage of genes with greater expression in men and women (3.11D)
```{r}
#change men vs women
perform_t_tests_all_genesmax <- function(vmen, vdifwomen) {
  # Get all gene names and desired order
  all_genes <- intersect(rownames(mitomen), rownames(mitodifwomen))
      desired_order <- c(5,2,4,7,6,3,8,9,1)
      
  # Initialize a matrix to store t-test results for each gene
 t_stat_df <- data.frame(gene_name = character(length(all_genes)),
                          t_statistic = numeric(length(all_genes)),
                          p_value = numeric(length(all_genes)),
                          avg_max_expression_men = numeric(length(all_genes)),
                          avg_max_expression_women = numeric(length(all_genes)),
                          stringsAsFactors = FALSE)
  
  # Initialize a counter variable for row index in t_stat_matrix
  row_index <- 1
  
  # Loop through each gene
 for (i in 1:length(all_genes)) {
    gene_name <- all_genes[i]
    
    # extract expression values and identify max
    geneid1 <- as.matrix(tapply(vmen$E[gene_name, ], list(sampleidmen, groupmen), mean))
    geneid1  <- geneid1 [, desired_order]
    geneid1[, 1:9] <- geneid1[, 1:9] - geneid1[, 1]
    geneid1 <- geneid1[, 2:9]
    geneid1 <- geneid1[rowSums(geneid1 > 0) > 0, , drop = FALSE]
    geneid1 <- apply(geneid1, 1, max)
    
    geneid2 <- as.matrix(tapply(vdifwomen$E[gene_name, ], list(sampleiddifwomen, groupdifwomen), mean))
    geneid2 <- geneid2 [, desired_order]
    geneid2 <- geneid2[, 1:9] - geneid2[, 1]
    geneid2 <- geneid2[, 2:9]
    geneid2 <- geneid2[rowSums(geneid2 > 0) > 0, , drop = FALSE]
    geneid2 <- apply(geneid2, 1, max)
    
    # Perform the t-test
    t_test_result <- t.test(geneid1, geneid2)
    
    # Store t-test result for the current gene in the data frame
    t_stat_df[i, ] <- list(gene_name = gene_name,
                           t_statistic = t_test_result$statistic,
                           p_value = t_test_result$p.value,
                           average_of_men = t_test_result$estimate[[1]],
                           average_of_women = t_test_result$estimate[[2]])
  }
  
  return(t_stat_df)
}

# Call the function with your data
t_test_all_genes <- perform_t_tests_all_genesmax(vmen, vdifwomen)
t_test_all_genespval <- subset(t_test_all_genes, t_test_all_genes$p_value<0.05)

higher_men_upreg <- t_test_all_genespval[t_test_all_genespval$avg_max_expression_men > t_test_all_genespval$avg_max_expression_women, ]
higher_women_upreg<- t_test_all_genespval[t_test_all_genespval$avg_max_expression_women > t_test_all_genespval$avg_max_expression_men, ]

menmaxup <- as.numeric(nrow(higher_men_upreg))
menmaxupgenes <- higher_men_upreg$gene_name
womenmaxup <- as.numeric(nrow(higher_women_upreg))
womenmaxupgenes <- higher_women_upreg$gene_name
totalmaxup <- as.numeric(nrow(t_test_all_genes))-(menmaxup + womenmaxup)

sexmax <- data.frame(
  category = c("Higher in men", "Higher in women", "No difference"),
  value = c(menmaxup, womenmaxup, totalmaxup)
)

category_colorsUP <- c("#94e2eb", "#e3b3e0","lightgrey")
category_colorsDOWN <- c("#34d8eb", "#e887e3","lightgrey")

sexmax$value_percentage <- (sexmax$value / sum(sexmax$value)) * 100

ggplot(sexmax, aes(x = "", y = value_percentage, fill = category)) +
  geom_bar(width = 1, stat = "identity", colour = "black") +
  scale_fill_manual(values = category_colorsUP, 
                    labels = paste0(sexmax$category, " (", round(sexmax$value_percentage, 1), "%)")) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = "Peak upregulation of differentially \n  expressed mitochondrial genes") +
  geom_text(aes(label = paste0(round(value_percentage, 1), "%")),
            position = position_stack(vjust = 0.5),
            size = 0) +
  theme(legend.text = element_text(size = 12),
        plot.title = element_text(size = 14)) +
  labs(fill = NULL)

#looking at min regulation 
  
perform_t_tests_all_genesmin <- function(vmen, vdifwomen) {
  # Get all gene names and desired order
  all_genes <- intersect(rownames(mitomen), rownames(mitodifwomen))
      desired_order <- c(5,2,4,7,6,3,8,9,1)
      
  # Initialize a matrix to store t-test results for each gene
 t_stat_df <- data.frame(gene_name = character(length(all_genes)),
                          t_statistic = numeric(length(all_genes)),
                          p_value = numeric(length(all_genes)),
                          avg_min_expression_men = numeric(length(all_genes)),
                          avg_min_expression_women = numeric(length(all_genes)),
                          stringsAsFactors = FALSE)
  
  # Initialize a counter variable for row index in t_stat_matrix
  row_index <- 1
  
  # Loop through each gene
 for (i in 1:length(all_genes)) {
    gene_name <- all_genes[i]
    
    # extract expression values and identify min
    geneid1 <- as.matrix(tapply(vmen$E[gene_name, ], list(sampleidmen, groupmen), mean))
    geneid1  <- geneid1 [, desired_order]
    geneid1[, 1:9] <- geneid1[, 1:9] - geneid1[, 1]
    geneid1 <- geneid1[, 2:9]
    geneid1 <- geneid1[rowSums(geneid1 < 0) > 0, , drop = FALSE]
    geneid1 <- apply(geneid1, 1, min)
    
    geneid2 <- as.matrix(tapply(vdifwomen$E[gene_name, ], list(sampleiddifwomen, groupdifwomen), mean))
    geneid2 <- geneid2 [, desired_order]
    geneid2 <- geneid2[, 1:9] - geneid2[, 1]
    geneid2 <- geneid2[, 2:9]
    geneid2 <- geneid2[rowSums(geneid2 < 0) > 0, , drop = FALSE]
    geneid2 <- apply(geneid2, 1, min)
    
    # Perform the t-test
    t_test_result <- t.test(geneid1, geneid2)
    
    # Store t-test result for the current gene in the data frame
    t_stat_df[i, ] <- list(gene_name = gene_name,
                           t_statistic = t_test_result$statistic,
                           p_value = t_test_result$p.value,
                           average_of_men = t_test_result$estimate[[1]],
                           average_of_women = t_test_result$estimate[[2]])
  }
  
  return(t_stat_df)
}

t_test_results_all_genes <- perform_t_tests_all_genesmin(vmen, vdifwomen)
t_test_all_genespval <- subset(t_test_results_all_genes, t_test_results_all_genes$p_value<0.05)

higher_men_downreg <- t_test_all_genespval[t_test_all_genespval$avg_min_expression_men < t_test_all_genespval$avg_min_expression_women, ]
higher_women_downreg<- t_test_all_genespval[t_test_all_genespval$avg_min_expression_women < t_test_all_genespval$avg_min_expression_men, ]

menminup <- as.numeric(nrow(higher_men_downreg))
menminupgenes <- higher_men_downreg$gene_name
womenminup <- as.numeric(nrow(higher_women_downreg))
womenminupgenes <- higher_women_downreg$gene_name
totalminup <- as.numeric(nrow(t_test_all_genes)-(menminup + womenminup))

sexmin <- data.frame(
  category = c("Lower in men", "Lower in women", "No difference"),
  value = c(menminup, womenminup, totalminup)
)

category_colorsUP <- c("#94e2eb", "#e3b3e0","lightgrey")
category_colorsDOWN <- c("#34d8eb", "#e887e3","lightgrey")

sexmin$value_percentage <- (sexmin$value / sum(sexmin$value)) * 100

ggplot(sexmin, aes(x = "", y = value_percentage, fill = category)) +
  geom_bar(width = 1, stat = "identity", colour = "black") +
  scale_fill_manual(values = category_colorsDOWN, 
                    labels = paste0(sexmin$category, " (", round(sexmin$value_percentage, 1), "%)")) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = "Peak downregulation of differentially \n  expressed mitochondrial genes") +
  geom_text(aes(label = paste0(round(value_percentage, 1), "%")),
            position = position_stack(vjust = 0.5),
            size = 0) +
  theme(legend.text = element_text(size = 12),
        plot.title = element_text(size = 14)) +
  labs(fill = NULL)

greater_men <- c(menmaxupgenes, menminupgenes)
greater_women <- c(womenmaxupgenes, womenminupgenes)

allsexDEGs <-c(row.names(mitomen), row.names(mitodifwomen))
allsexDEGS <- unique(allsexDEGs)

womenhighermitogenesenrich <- enricher(greater_women, universe = allsexDEGs, pAdjustMethod = "BH", minGSSize = 1, gson = NULL, TERM2GENE = mitocartaadj)
womenhighermitogenesenrich <- as.data.frame(womenhighermitogenesenrich@result) %>% arrange(pvalue) %>% mutate(Sex = "Greater amplitude in women")

menhighermitogenesenrich <- enricher(greater_men, universe = allsexDEGs, pAdjustMethod = "BH", minGSSize = 1, gson = NULL, TERM2GENE = mitocartaadj)
menhighermitogenesenrich <- as.data.frame(menhighermitogenesenrich@result) %>% arrange(pvalue) %>% mutate(Sex = "Greater amplitude in men")

print(womenmaxupgenes)
print(womenminupgenes)
print(menmaxupgenes)
print(menminupgenes)


perform_t_tests_all_genesAUC <- function(vmen, vdifwomen) {
  # Get all gene names and desired order
  all_genes <- intersect(rownames(mitomen), rownames(mitodifwomen))
      desired_order <- c(5,2,4,7,6,3,8,9,1)
      
  # Initialize a matrix to store t-test results for each gene
 t_stat_df <- data.frame(gene_name = character(length(all_genes)),
                          t_statistic = numeric(length(all_genes)),
                          p_value = numeric(length(all_genes)),
                          avg_AUC_expression_men = numeric(length(all_genes)),
                          avg_AUC_expression_women = numeric(length(all_genes)),
                          stringsAsFactors = FALSE)
 
  calculate_auc_trapz <- function(row) {
      x <- as.numeric(c("0", "3", "6", "9", "12", "24", "48"))
      y <- as.numeric(row)                
      auc <- trapz(x, y)
      return(auc)
    }

  # Initialize a counter variable for row index in t_stat_matrix
  row_index <- 1
  
  # Loop through each gene
 for (i in 1:length(all_genes)) {
    gene_name <- all_genes[i]
    
    # extract expression values and identify AUC
    geneid1 <- as.matrix(tapply(vmen$E[gene_name, ], list(sampleidmen, groupmen), mean))
    geneid1  <- geneid1 [, desired_order]
    geneid1[, 1:9] <- geneid1[, 1:9] - geneid1[, 1]
    geneid1 <- geneid1[, 3:9]
    rows_to_remove1 <- c(15)
    geneid1 <- geneid1[-rows_to_remove1, ]
    auc1 <- apply(geneid1, 1, calculate_auc_trapz)
    
    geneid2 <- as.matrix(tapply(vdifwomen$E[gene_name, ], list(sampleiddifwomen, groupdifwomen), mean))
    geneid2 <- geneid2 [, desired_order]
    geneid2 <- geneid2[, 1:9] - geneid2[, 1]
    geneid2 <- geneid2[, 3:9]
    rows_to_remove2 <- c(6)
    geneid2 <- geneid2[-rows_to_remove2, ]
    auc2 <- apply(geneid2, 1, calculate_auc_trapz)
    
    # Perform the t-test
    t_test_result <- t.test(auc1, auc2)
    
    # Store t-test result for the current gene in the data frame
    t_stat_df[i, ] <- list(gene_name = gene_name,
                           t_statistic = t_test_result$statistic,
                           p_value = t_test_result$p.value,
                           auc_of_men = t_test_result$estimate[[1]],
                           auc_of_women = t_test_result$estimate[[2]])
  }
  
  return(t_stat_df)
}

t_test_results_all_genes <- perform_t_tests_all_genesAUC(vmen, vdifwomen)
t_test_all_genespval <- subset(t_test_results_all_genes, t_test_results_all_genes$p_value<0.05)

higher_men_downreg <- t_test_all_genespval[t_test_all_genespval$avg_AUC_expression_men < t_test_all_genespval$avg_AUC_expression_women, ]
higher_women_downreg<- t_test_all_genespval[t_test_all_genespval$avg_AUC_expression_women < t_test_all_genespval$avg_AUC_expression_men, ]

menAUCup <- as.numeric(nrow(higher_men_downreg))
menAUCupgenes <- higher_men_downreg$gene_name
womenAUCup <- as.numeric(nrow(higher_women_downreg))
womenAUCupgenes <- higher_women_downreg$gene_name
totalAUCup <- as.numeric(nrow(t_test_all_genes)-(menAUCup + womenAUCup))

sexAUC <- data.frame(
  category = c("Larger in men", "Larger in women", "No difference"),
  value = c(menAUCup, womenAUCup, totalAUCup)
)

category_colorsUP <- c("#94e2eb", "#e3b3e0","lightgrey")
category_colorsDOWN <- c("#34d8eb", "#e887e3","lightgrey")

sexAUC$value_percentage <- (sexAUC$value / sum(sexAUC$value)) * 100

ggplot(sexAUC, aes(x = "", y = value_percentage, fill = category)) +
  geom_bar(width = 1, stat = "identity", colour = "black") +
  scale_fill_manual(values = category_colorsDOWN, 
                    labels = paste0(sexAUC$category, " (", round(sexAUC$value_percentage, 1), "%)")) +
  coord_polar(theta = "y") +
  theme_void() +
  labs(title = "AUC of differentially \n  expressed mitochondrial genes") +
  geom_text(aes(label = paste0(round(value_percentage, 1), "%")),
            position = position_stack(vjust = 0.5),
            size = 0) +
  theme(legend.text = element_text(size = 12),
        plot.title = element_text(size = 14)) +
  labs(fill = NULL)

```

#enrichment analysis of genes only DE in one sex or with greater expression (3.11E)
```{r}

allsexDEgenes <- c(greater_men, greater_women, notDEexpresseddifwomen, notDEexpressedmen)

sexspecificmitogenesenrich <- enricher(allsexDEgenes, universe = allsexDEGs, pAdjustMethod = "BH", minGSSize = 1, gson = NULL, TERM2GENE = mitocartaadj)
sexspecificmitogenesenrich <- as.data.frame(sexspecificmitogenesenrich@result) %>% arrange(pvalue) %>% mutate(Sex = "All sex-specific DEGs")

sexspecificmitogenes <- rbind(menhighermitogenesenrich, womenhighermitogenesenrich, mitocartaenrichonlymen, mitocartaenrichonlydifwomen, sexspecificmitogenesenrich)

sexspecificmitogenes$Description <- fct_rev(sexspecificmitogenes$Description)

sexspecificmitogenes$Sex <- as.character(sexspecificmitogenes$Sex)
sexspecificmitogenes$Sex <- factor(sexspecificmitogenes$Sex, levels = unique(sexspecificmitogenes$Sex))

sexstrip_background2 <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      Men = "#02e1f5",
      Women = "#f02bd9",
      Men = "#02e1f5",
      Women = "#f02bd9",
      Shared = "white"
    )
  )
)

ggplot(sexspecificmitogenes, aes(x= Count, y = Description, fill = p.adjust)) +
  geom_col() +
  scale_size_continuous(name = "Count", range = c(1, 7)) +
  scale_fill_gradientn(name = "Adjusted p-value", colours = c("red", "blue"), limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01), na.value = "black") +
  scale_x_continuous(breaks = seq(0, max(sexspecificmitogenes$Count), by = 5)) +
  theme_minimal() +
  labs(y=NULL, x = "Number of genes") +
  ggtitle(bquote("Mitochondrial genes with sex-specific regulation")) +
  theme(panel.border = element_rect(color = "black", fill = NA),
        strip.text.y = element_text(angle = 90, size = 10, face="bold"),
        legend.text = element_text(size = 6),
                axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
    axis.ticks.length = unit(0.1, "cm"),
        strip.background = element_rect(fill = "lightgrey"),
        panel.spacing = unit(0, "cm"),
     strip.placement = "outside",
    legend.position = "bottom",
     legend.justification = "center"
  ) +
  guides(size = guide_legend(), color = guide_colorbar(title.position = "right")) +
    facet_nested(.~Sex, scales = "free_y", switch = "y", space = "free_y", strip=sexstrip_background2)
```

#mitochondrial clustering for men (for 3.11C)
```{r}
mitoDEsoftmen <- mitomen[c(1:8)]
colnames(mitoDEsoftmen) <- c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")
mitoDEsoftmen <- as.matrix(mitoDEsoftmen)
tmp_exprmen <- new('ExpressionSet', exprs = mitoDEsoftmen)

#estimation of 'fuzzifier' constant
m1men <- mestimate(tmp_exprmen)

#estimation and visualisation of number of clusters (no super clean elbow)
Dmin(tmp_exprmen,m1men,crange=seq(2,30,2),repeats=2,visu=TRUE)

#Soft clustering of data
fuzzclustermen <- mfuzz(tmp_exprmen, centers = 6, m = m1men )
#Plot of soft clustering of data
mfuzz.plot2(tmp_exprmen, cl = fuzzclustermen, mfrow = c(2, 3), colo="fancy", min.mem = 0.27, time.labels = colnames(mitoDEsoftmen), xlab="Time", ylab =(Log[2]*FC~ "relative to PRE"), ax.col="black",bg = "white",col.axis="black",col.lab="black",
                        col.main="black",col.sub="black",col="black",
                        centre.col="green",centre.lwd=3, centre=TRUE, x11=FALSE)

mfuzzColorBar(col="fancy", horizontal=FALSE, main="Gene membership values", cex.main=0.8)

overlapfcmen <- Mfuzz::overlap(fuzzclustermen)
overlap.plot(fuzzclustermen, overlapfcmen, thres=0.29)

#Determination of number of genes within cluster (only considers a genes highest membership - hard cluster for sake of identification)
fuzzclustermen$size

cluster_count <- data.frame(
  Total_Count = fuzzclustermen$size,
  Cluster = c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6"))

#generation of gene sets within all clusters
clusterprofilemen <- acore(tmp_exprmen,fuzzclustermen, min.acore=0.24)

#selection of genes within a specific cluster (CLUSTER generation betwen men and women will be different - will need to adjust each to match the clusters from whole time course)
cluster1men <- clusterprofilemen[[1]]
genes_to_test1men <- cluster1men$NAME

cluster2men <- clusterprofilemen[[2]]
genes_to_test2men <- cluster2men$NAME

cluster3men <- clusterprofilemen[[3]]
genes_to_test3men <- cluster3men$NAME

cluster4men <- clusterprofilemen[[4]]
genes_to_test4men <- cluster4men$NAME

cluster5men <- clusterprofilemen[[5]]
genes_to_test5men <- cluster5men$NAME

cluster6men <- clusterprofilemen[[6]]
genes_to_test6men <- cluster6men$NAME

membershipmen <- fuzzclustermen$membership
colnames(membershipmen) <- c("one", "two", "three", "four", "five", "six")

create_cluster_plot <- function(gene_cluster_list, membership_column, cluster_name) {
  cluster_data <- rna_compMEN[rownames(rna_compMEN) %in% gene_cluster_list, ]
  cluster_data <- cluster_data[, 1:8]
  cluster_data <- merge(cluster_data, membershipmen, by.x = 0, by.y = 0)
  colnames(cluster_data) <- c("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", "one", "two", "three", "four", "five", "six")
  
  cluster_data <- cluster_data %>% dplyr::select("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", membership_column)

  cluster_plot_data <- gather(cluster_data, key = "Time", value = "LogFC", -Gene, -{{membership_column}})
  cluster_plot_data$Time <- as.character(cluster_plot_data$Time)
  cluster_plot_data$Time <- factor(cluster_plot_data$Time, levels = unique(cluster_plot_data$Time))
  cluster_plot_data <- na.omit(cluster_plot_data)
  
  cluster_plot <- ggplot(cluster_plot_data, aes(y=LogFC, x=Time, group=Gene, color=!!sym(membership_column))) + 
    labs(x = "Time", y = bquote(Log[2] * FC ~ "relative to PRE"))  +
    geom_line(data = cluster_plot_data, size = 0.3) +
scale_color_gradientn(name = "Cluster membership", colours = c("yellow", "navy"), limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4), guide = "none") + 
    stat_summary(aes(group = 1), geom = "line", fun = "mean", colour = "green", size = 1.5) +
    ggtitle(cluster_name) +
    theme_set(theme_bw())
  
  return(cluster_plot)
}

cluster1men_plot <- create_cluster_plot(genes_to_test1men, "one", "Cluster 1")
cluster2men_plot <- create_cluster_plot(genes_to_test2men, "two", "Cluster 2")
cluster3men_plot <- create_cluster_plot(genes_to_test3men, "three", "Cluster 3")
cluster4men_plot <- create_cluster_plot(genes_to_test6men, "six", "Cluster 4")
cluster5men_plot <- create_cluster_plot(genes_to_test5men, "five", "Cluster 5")
cluster6men_plot <- create_cluster_plot(genes_to_test4men, "four", "Cluster 6")

membership_legend <- ggplot(data.frame(value = c(0, 1)), aes(x = 1, y = value, color = value)) +
  geom_point(size = 0) +
  scale_color_gradientn(name = "Cluster membership", colours = c("yellow", "navy"), limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4)) +
  theme_void() +
  theme(legend.position = "right")

grid.arrange(cluster1men_plot, cluster2men_plot, cluster3men_plot, membership_legend, cluster4men_plot, cluster5men_plot, cluster6men_plot, ncol = 4, widths = c(3, 3, 3, 1.5))

```

#mitochondrial clustering for women (for 3.11C)
```{r}
mitoDEsoftdifwomen <- mitodifwomen[c(1:8)]
colnames(mitoDEsoftdifwomen) <- c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")
mitoDEsoftdifwomen <- as.matrix(mitoDEsoftdifwomen)
tmp_exprdifwomen <- new('ExpressionSet', exprs = mitoDEsoftdifwomen)

#estimation of 'fuzzifier' constant
m1difwomen <- mestimate(tmp_exprdifwomen)

#estimation and visualisation of number of clusters (no super clean elbow)
Dmin(tmp_exprdifwomen,m1difwomen,crange=seq(2,30,2),repeats=2,visu=TRUE)

#Soft clustering of data
fuzzclusterdifwomen <- mfuzz(tmp_exprdifwomen, centers = 6, m = m1difwomen )
#Plot of soft clustering of data
mfuzz.plot2(tmp_exprdifwomen, cl = fuzzclusterdifwomen, mfrow = c(2, 3), colo="fancy", min.mem = 0.26, time.labels = colnames(mitoDEsoftdifwomen), xlab="Time", ylab =(Log[2]*FC~ "relative to PRE"), ax.col="black",bg = "white",col.axis="black",col.lab="black",
                        col.main="black",col.sub="black",col="black",
                        centre.col="green",centre.lwd=3, centre=TRUE, x11=FALSE)

mfuzzColorBar(col="fancy", horizontal=FALSE, main="Gene membership values", cex.main=0.8)

overlapfcdifwomen <- Mfuzz::overlap(fuzzclusterdifwomen)
overlap.plot(fuzzclusterdifwomen, overlapfcdifwomen, thres=0.29)

#Determination of number of genes within cluster (only considers a genes highest membership - hard cluster for sake of identification lol)
fuzzclusterdifwomen$size

cluster_count <- data.frame(
  Total_Count = fuzzclusterdifwomen$size,
  Cluster = c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6"))

#generation of gene sets within all clusters
clusterprofiledifwomen <- acore(tmp_exprdifwomen,fuzzclusterdifwomen, min.acore=0.24)

#selection of genes within a specific cluster (CLUSTER generation betwen difwomen and wodifwomen will be different - will need to adjust each to match the clusters from whole time course)
cluster1difwomen <- clusterprofiledifwomen[[1]]
genes_to_test1difwomen <- cluster1difwomen$NAME

cluster2difwomen <- clusterprofiledifwomen[[2]]
genes_to_test2difwomen <- cluster2difwomen$NAME

cluster3difwomen <- clusterprofiledifwomen[[3]]
genes_to_test3difwomen <- cluster3difwomen$NAME

cluster4difwomen <- clusterprofiledifwomen[[4]]
genes_to_test4difwomen <- cluster4difwomen$NAME

cluster5difwomen <- clusterprofiledifwomen[[5]]
genes_to_test5difwomen <- cluster5difwomen$NAME

cluster6difwomen <- clusterprofiledifwomen[[6]]
genes_to_test6difwomen <- cluster6difwomen$NAME

membershipdifwomen <- fuzzclusterdifwomen$membership
colnames(membershipdifwomen) <- c("one", "two", "three", "four", "five", "six")

create_cluster_plot <- function(gene_cluster_list, membership_column, cluster_name) {
  cluster_data <- rna_compdifwomen[rownames(rna_compdifwomen) %in% gene_cluster_list, ]
  cluster_data <- cluster_data[, 1:8]
  cluster_data <- merge(cluster_data, membershipdifwomen, by.x = 0, by.y = 0)
  colnames(cluster_data) <- c("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", "one", "two", "three", "four", "five", "six")
  
  cluster_data <- cluster_data %>% dplyr::select("Gene", "MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H", membership_column)

  cluster_plot_data <- gather(cluster_data, key = "Time", value = "LogFC", -Gene, -{{membership_column}})
  cluster_plot_data$Time <- as.character(cluster_plot_data$Time)
  cluster_plot_data$Time <- factor(cluster_plot_data$Time, levels = unique(cluster_plot_data$Time))
  cluster_plot_data <- na.omit(cluster_plot_data)
  
  cluster_plot <- ggplot(cluster_plot_data, aes(y=LogFC, x=Time, group=Gene, color=!!sym(membership_column))) + 
    labs(x = "Time", y = bquote(Log[2] * FC ~ "relative to PRE"))  +
    geom_line(data = cluster_plot_data, size = 0.3) +
scale_color_gradientn(name = "Cluster membership", colours = c("yellow", "navy"), limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4), guide = "none") + 
    stat_summary(aes(group = 1), geom = "line", fun = "mean", colour = "green", size = 1.5) +
    ggtitle(cluster_name) +
    theme_set(theme_bw())
  
  return(cluster_plot)
}

cluster1difwomen_plot <- create_cluster_plot(genes_to_test1difwomen, "one", "Cluster 1")
cluster2difwomen_plot <- create_cluster_plot(genes_to_test3difwomen, "three", "Cluster 2")
cluster3difwomen_plot <- create_cluster_plot(genes_to_test5difwomen, "five", "Cluster 3")
cluster4difwomen_plot <- create_cluster_plot(genes_to_test6difwomen, "six", "Cluster 4")
cluster5difwomen_plot <- create_cluster_plot(genes_to_test4difwomen, "four", "Cluster 5")
cluster6difwomen_plot <- create_cluster_plot(genes_to_test2difwomen, "two", "Cluster 6")

membership_legend <- ggplot(data.frame(value = c(0, 1)), aes(x = 1, y = value, color = value)) +
  geom_point(size = 0) +
  scale_color_gradientn(name = "Cluster membership", colours = c("yellow", "navy"), limits = c(0, 1), breaks = seq(0.2, 1, by = 0.4)) +
  theme_void() +
  theme(legend.position = "right")

grid.arrange(cluster1difwomen_plot, cluster2difwomen_plot, cluster3difwomen_plot, membership_legend, cluster4difwomen_plot, cluster5difwomen_plot, cluster6difwomen_plot, ncol = 4, widths = c(3, 3, 3, 1.5))
```

#mitochondrial pathway enrichment for mito clusters in men and women (did not use)
```{r}
mitocartaadj <- read.xlsx("data/mitocarta30adj.xlsx")
colnames(mitocartaadj) <- c("term", "gene", "hier") #was trying to do some fancy facetwrapping for hier but didn't get it working how I want yet...
mitocartaadj <- mitocartaadj %>%
  mutate(gene = strsplit(as.character(gene), ", ")) %>%
  unnest(cols = gene)
womenmitoDEGs <- rownames(mitoDEsoftdifwomen)
menmitoDEGs <- rownames(mitoDEsoftmen)

mitoclustersexenrichment <- function(genes_to_test_list, sex, sexmitoDEGs) {
  result_list <- lapply(seq_along(genes_to_test_list), function(i) {
    enrich_result <- enricher(genes_to_test_list[[i]], universe = sexmitoDEGs, pAdjustMethod = "BH", minGSSize = 1, gson = NULL, TERM2GENE = mitocartaadj)
    df_result <- as.data.frame(enrich_result@result) %>%
      arrange(pvalue) %>%
      mutate(Condition = paste("Cluster", i), Sex = sex)
    return(df_result)
  })
  combined_results <- do.call(rbind, result_list)
  return(combined_results)}

cluster1men <- clusterprofilemen[[1]]
genes_to_test1men <- cluster1men$NAME

cluster2men <- clusterprofilemen[[2]]
genes_to_test2men <- cluster2men$NAME

cluster3men <- clusterprofilemen[[3]]
genes_to_test3men <- cluster3men$NAME

cluster4men <- clusterprofilemen[[4]]
genes_to_test6men <- cluster4men$NAME

cluster5men <- clusterprofilemen[[5]]
genes_to_test5men <- cluster5men$NAME

cluster6men <- clusterprofilemen[[6]]
genes_to_test4men <- cluster6men$NAME

combinedMCenrichmentmen <- mitoclustersexenrichment(list(genes_to_test1men, genes_to_test2men, genes_to_test3men, genes_to_test4men, genes_to_test5men, genes_to_test6men), "Men", menmitoDEGs)

cluster1difwomen <- clusterprofiledifwomen[[1]]
genes_to_test1difwomen <- cluster1difwomen$NAME

cluster2difwomen <- clusterprofiledifwomen[[2]]
genes_to_test6difwomen <- cluster2difwomen$NAME

cluster3difwomen <- clusterprofiledifwomen[[3]]
genes_to_test2difwomen <- cluster3difwomen$NAME

cluster4difwomen <- clusterprofiledifwomen[[4]]
genes_to_test5difwomen <- cluster4difwomen$NAME

cluster5difwomen <- clusterprofiledifwomen[[5]]
genes_to_test3difwomen <- cluster5difwomen$NAME

cluster6difwomen <- clusterprofiledifwomen[[6]]
genes_to_test4difwomen <- cluster6difwomen$NAME

cluster1difwomen_plot <- create_cluster_plot(genes_to_test1difwomen, "one", "Cluster 1")
cluster2difwomen_plot <- create_cluster_plot(genes_to_test3difwomen, "three", "Cluster 2")
cluster3difwomen_plot <- create_cluster_plot(genes_to_test5difwomen, "five", "Cluster 3")
cluster4difwomen_plot <- create_cluster_plot(genes_to_test6difwomen, "six", "Cluster 4")
cluster5difwomen_plot <- create_cluster_plot(genes_to_test4difwomen, "four", "Cluster 5")
cluster6difwomen_plot <- create_cluster_plot(genes_to_test2difwomen, "two", "Cluster 6")

combinedMCenrichmentdifwomen<- mitoclustersexenrichment(list(genes_to_test1difwomen, genes_to_test2difwomen, genes_to_test3difwomen, genes_to_test4difwomen, genes_to_test5difwomen, genes_to_test6difwomen), "Women", womenmitoDEGs)

combined_resultsMCmenvsdifwomen <- rbind(combinedMCenrichmentdifwomen, combinedMCenrichmentmen)

combined_resultsMCmenvsdifwomen$GeneRatio <- sapply(strsplit(combined_resultsMCmenvsdifwomen$GeneRatio, "/"), function(x) as.numeric(x[1])/as.numeric(x[2]))

condition_order2 <- c("Cluster 1", "Cluster 2", "Cluster 3", "Cluster 4", "Cluster 5", "Cluster 6")
combined_resultsMCmenvsdifwomen$Condition <- factor(combined_resultsMCmenvsdifwomen$Condition, levels = condition_order2)

combined_resultsMCmenvsdifwomen$Description <- fct_rev(combined_resultsMCmenvsdifwomen$Description)

sexstrip_background <- strip_nested(
  text_x = elem_list_text(colour = "black", face = "bold"),
  background_x = elem_list_rect(
    fill = c(
      Men = "#02e1f5",
      Women = "#f02bd9",
      Shared = "#e8bc74"
    )
  )
)

ggplot(combined_resultsMCmenvsdifwomen, aes(x = Condition, y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_size_continuous(name = "Count", range = c(1, 5)) +
  scale_color_gradientn(name = "Adjusted p-value", colours = c("red", "blue"), limits = c(0, 0.05), breaks = seq(0.01, 0.05, by = 0.01), na.value = "black") +
  theme_minimal() +
  labs(y=NULL) +
  ggtitle("Sex-specific mitochondrial cluster enrichment (MitoCarta 3.0)") +
  theme(panel.border = element_rect(color = "black", fill = NA),
        strip.text.y = element_text(angle = 90, size = 8, face="bold"),
         axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        legend.text = element_text(size = 8),
                axis.ticks.y = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
    axis.ticks.length = unit(0.1, "cm"),
        strip.background = element_rect(fill = "lightgrey"),
        panel.spacing = unit(0, "cm"),
     strip.placement = "outside"
  ) +
  guides(size = guide_legend(), color = guide_colorbar(title.position = "top")) +
   facet_nested(.~Sex, scales = "free_y", switch = "y", space = "free_y", strip = sexstrip_background)
```

#creation of combined mitochondrial gene cluster figure (3.11C)
```{r}
cluster1 <- openxlsx::read.xlsx("data/cluster1.xlsx")
genes_to_test1 <- cluster1$NAME
cluster2 <- openxlsx::read.xlsx("data/cluster2.xlsx")
genes_to_test2 <- cluster2$NAME
cluster3 <- openxlsx::read.xlsx("data/cluster3.xlsx")
genes_to_test3 <- cluster3$NAME
cluster4 <- openxlsx::read.xlsx("data/cluster4.xlsx")
genes_to_test4 <- cluster4$NAME
cluster5 <- openxlsx::read.xlsx("data/cluster5.xlsx")
genes_to_test5 <- cluster5$NAME
cluster6 <- openxlsx::read.xlsx("data/cluster6.xlsx")
genes_to_test6 <- cluster6$NAME

sexclusterdata <- function(rna_comp, rna_comp_men, rna_comp_dif_women) {
  clusters <- list(Cluster1 = genes_to_test1,
                   Cluster2 = genes_to_test2,
                   Cluster3 = genes_to_test3,
                   Cluster4 = genes_to_test4,
                   Cluster5 = genes_to_test5,
                   Cluster6 = genes_to_test6)
  
  all_cluster_data <- list()
  
  for (i in 1:length(clusters)) {
    men_data <- rna_comp_men %>%
      dplyr::filter(row.names(rna_comp_men) %in% clusters[[i]]) %>%
      dplyr::select(1:8) %>% 
      `colnames<-`(c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")) %>%
      mutate(Sex = "Men", Cluster = paste0("Cluster ", i))
    
    women_data <- rna_comp_dif_women %>%
     dplyr::filter(row.names(rna_comp_dif_women) %in% clusters[[i]]) %>%
      dplyr::select(1:8) %>%
      `colnames<-`(c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")) %>%
      mutate(Sex = "Women", Cluster = paste0("Cluster ", i))
    
    together_data <- rna_comp %>%
     dplyr::filter(row.names(rna_comp) %in% clusters[[i]]) %>%
     dplyr::select(1:8) %>%
      `colnames<-`(c("MID", "POST", "3H", "6H", "9H", "12H", "24H", "48H")) %>%
      mutate(Sex = "Together", Cluster = paste0("Cluster ", i))
    
    # Combine the three data frames
    all_cluster_data[[i]] <- bind_rows(together_data, men_data, women_data)
  }
  
  return(all_cluster_data)
}


sexclusterdatadata <- sexclusterdata(rna_comp, rna_compMEN, rna_compdifwomen)
Cluster1sexplot <- sexclusterdatadata[[1]]
Cluster2sexplot <- sexclusterdatadata[[2]]
Cluster3sexplot <- sexclusterdatadata[[3]]
Cluster4sexplot <- sexclusterdatadata[[4]]
Cluster5sexplot <- sexclusterdatadata[[5]]
Cluster6sexplot <- sexclusterdatadata[[6]]

sexplots <- function(Clustersexplot, Clustername) {
  
  Clusterplot <- Clustersexplot %>% 
    rownames_to_column(var = "GeneID") %>% 
    gather(Time, LogFC, -GeneID, -Cluster, -Sex)
  
  mengenenumber <- Clusterplot %>% 
    dplyr::filter(Sex == "Men") %>% 
    summarize(unique_genes = as.character(n_distinct(GeneID))) %>% 
    pull(unique_genes)
  
  womengenenumber <- Clusterplot %>% 
    dplyr::filter(Sex == "Women") %>% 
    summarize(unique_genes = as.character(n_distinct(GeneID))) %>% 
    pull(unique_genes)
  
  togethergenenumber <- Clusterplot %>% 
    dplyr::filter(Sex == "Together") %>% 
    summarize(unique_genes = as.character(n_distinct(GeneID))) %>% 
    pull(unique_genes)
  
  Clusterplot$Time <- as.character(Clusterplot$Time)
  Clusterplot$Time <- factor(Clusterplot$Time, levels = unique(Clusterplot$Time))
  
  mean_data <- Clusterplot %>%
    group_by(Time, Sex) %>%
    summarise(mean_LogFC = mean(LogFC),
              stdev = sd(LogFC))
  
  mean_data$Sex <- factor(mean_data$Sex, levels = c("Men", "Women", "Together"))

  
  p <- ggplot(mean_data, aes(x = Time, y = mean_LogFC, group = Sex, colour = Sex)) + 
    geom_line(size = 1) +
    geom_point(size = 3, colour = "black") + 
    ggtitle(paste0(Clustername, " (n = ", 
                   "<span style='color:#02e1f5'>", mengenenumber, "</span>, ",
                   "<span style='color:#f02bd9'>", womengenenumber, "</span>, ",
                   "<span style='color:#00FF00'>", togethergenenumber, "</span>)")) +
    labs(x = "Time", y = bquote(Log[2] * FC ~ "relative to PRE"), color = "Sex") +
    scale_colour_manual(values = c("Men" = "#02e1f5", "Women" = "#f02bd9", "Together" = "#00FF00"), guide = "none") +
    geom_errorbar(aes(ymin = mean_LogFC - stdev, ymax = mean_LogFC + stdev), width = .5,
                  position = position_dodge(.5)) +
    expand_limits(y = c(min(mean_data$mean_LogFC) - 0.1, max(mean_data$mean_LogFC) + 0.1)) +
    theme_bw() +
    theme(plot.title = element_markdown())
  
  return(p)
}

cluster1sexplot <- sexplots(Cluster1sexplot, "Cluster 1")
cluster2sexplot <- sexplots(Cluster2sexplot, "Cluster 2")
cluster3sexplot <- sexplots(Cluster3sexplot, "Cluster 3")
cluster4sexplot <- sexplots(Cluster4sexplot, "Cluster 4")
cluster5sexplot <- sexplots(Cluster5sexplot, "Cluster 5")
cluster6sexplot <- sexplots(Cluster6sexplot, "Cluster 6")

membership_legend2 <- ggplot(data.frame(value = c("Men", "Women", "Together")), aes(x = 1, y = value, color = value)) +
  geom_line(size = 6) +
  scale_color_manual(name = "Sex", values = c("Men" = "#02e1f5", "Women" = "#f02bd9", "Together" = "green"), limits = c("Men", "Women", "Together")) +
  theme_void() +
  theme(legend.position = "right")

grid.arrange(cluster1sexplot, cluster2sexplot, cluster3sexplot, membership_legend2, cluster4sexplot, cluster5sexplot, cluster6sexplot, ncol = 4, widths = c(3, 3, 3, 1.5))
```
